// @aws-sdk/client-route-53/dist-cjs/index.js
'use strict';

var middlewareHostHeader = require('@aws-sdk/middleware-host-header');
var middlewareLogger = require('@aws-sdk/middleware-logger');
var middlewareRecursionDetection = require('@aws-sdk/middleware-recursion-detection');
var middlewareUserAgent = require('@aws-sdk/middleware-user-agent');
var configResolver = require('@smithy/config-resolver');
var core = require('@smithy/core');
var schema = require('@smithy/core/schema');
var middlewareContentLength = require('@smithy/middleware-content-length');
var middlewareEndpoint = require('@smithy/middleware-endpoint');
var middlewareRetry = require('@smithy/middleware-retry');
var smithyClient = require('@smithy/smithy-client');
var httpAuthSchemeProvider = require('./auth/httpAuthSchemeProvider');
var runtimeConfig = require('./runtimeConfig');
var regionConfigResolver = require('@aws-sdk/region-config-resolver');
var protocolHttp = require('@smithy/protocol-http');
var middlewareSdkRoute53 = require('@aws-sdk/middleware-sdk-route53');
var utilWaiter = require('@smithy/util-waiter');

const resolveClientEndpointParameters = (options) => {
    return Object.assign(options, {
        useDualstackEndpoint: options.useDualstackEndpoint ?? false,
        useFipsEndpoint: options.useFipsEndpoint ?? false,
        defaultSigningName: "route53",
    });
};
const commonParams = {
    UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
    Endpoint: { type: "builtInParams", name: "endpoint" },
    Region: { type: "builtInParams", name: "region" },
    UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" },
};

const getHttpAuthExtensionConfiguration = (runtimeConfig) => {
    const _httpAuthSchemes = runtimeConfig.httpAuthSchemes;
    let _httpAuthSchemeProvider = runtimeConfig.httpAuthSchemeProvider;
    let _credentials = runtimeConfig.credentials;
    return {
        setHttpAuthScheme(httpAuthScheme) {
            const index = _httpAuthSchemes.findIndex((scheme) => scheme.schemeId === httpAuthScheme.schemeId);
            if (index === -1) {
                _httpAuthSchemes.push(httpAuthScheme);
            }
            else {
                _httpAuthSchemes.splice(index, 1, httpAuthScheme);
            }
        },
        httpAuthSchemes() {
            return _httpAuthSchemes;
        },
        setHttpAuthSchemeProvider(httpAuthSchemeProvider) {
            _httpAuthSchemeProvider = httpAuthSchemeProvider;
        },
        httpAuthSchemeProvider() {
            return _httpAuthSchemeProvider;
        },
        setCredentials(credentials) {
            _credentials = credentials;
        },
        credentials() {
            return _credentials;
        },
    };
};
const resolveHttpAuthRuntimeConfig = (config) => {
    return {
        httpAuthSchemes: config.httpAuthSchemes(),
        httpAuthSchemeProvider: config.httpAuthSchemeProvider(),
        credentials: config.credentials(),
    };
};

const resolveRuntimeExtensions = (runtimeConfig, extensions) => {
    const extensionConfiguration = Object.assign(regionConfigResolver.getAwsRegionExtensionConfiguration(runtimeConfig), smithyClient.getDefaultExtensionConfiguration(runtimeConfig), protocolHttp.getHttpHandlerExtensionConfiguration(runtimeConfig), getHttpAuthExtensionConfiguration(runtimeConfig));
    extensions.forEach((extension) => extension.configure(extensionConfiguration));
    return Object.assign(runtimeConfig, regionConfigResolver.resolveAwsRegionExtensionConfiguration(extensionConfiguration), smithyClient.resolveDefaultRuntimeConfig(extensionConfiguration), protocolHttp.resolveHttpHandlerRuntimeConfig(extensionConfiguration), resolveHttpAuthRuntimeConfig(extensionConfiguration));
};

class Route53Client extends smithyClient.Client {
    config;
    constructor(...[configuration]) {
        const _config_0 = runtimeConfig.getRuntimeConfig(configuration || {});
        super(_config_0);
        this.initConfig = _config_0;
        const _config_1 = resolveClientEndpointParameters(_config_0);
        const _config_2 = middlewareUserAgent.resolveUserAgentConfig(_config_1);
        const _config_3 = middlewareRetry.resolveRetryConfig(_config_2);
        const _config_4 = configResolver.resolveRegionConfig(_config_3);
        const _config_5 = middlewareHostHeader.resolveHostHeaderConfig(_config_4);
        const _config_6 = middlewareEndpoint.resolveEndpointConfig(_config_5);
        const _config_7 = httpAuthSchemeProvider.resolveHttpAuthSchemeConfig(_config_6);
        const _config_8 = resolveRuntimeExtensions(_config_7, configuration?.extensions || []);
        this.config = _config_8;
        this.middlewareStack.use(schema.getSchemaSerdePlugin(this.config));
        this.middlewareStack.use(middlewareUserAgent.getUserAgentPlugin(this.config));
        this.middlewareStack.use(middlewareRetry.getRetryPlugin(this.config));
        this.middlewareStack.use(middlewareContentLength.getContentLengthPlugin(this.config));
        this.middlewareStack.use(middlewareHostHeader.getHostHeaderPlugin(this.config));
        this.middlewareStack.use(middlewareLogger.getLoggerPlugin(this.config));
        this.middlewareStack.use(middlewareRecursionDetection.getRecursionDetectionPlugin(this.config));
        this.middlewareStack.use(core.getHttpAuthSchemeEndpointRuleSetPlugin(this.config, {
            httpAuthSchemeParametersProvider: httpAuthSchemeProvider.defaultRoute53HttpAuthSchemeParametersProvider,
            identityProviderConfigProvider: async (config) => new core.DefaultIdentityProviderConfig({
                "aws.auth#sigv4": config.credentials,
            }),
        }));
        this.middlewareStack.use(core.getHttpSigningPlugin(this.config));
    }
    destroy() {
        super.destroy();
    }
}

let Route53ServiceException$1 = class Route53ServiceException extends smithyClient.ServiceException {
    constructor(options) {
        super(options);
        Object.setPrototypeOf(this, Route53ServiceException.prototype);
    }
};

const AccountLimitType = {
    MAX_HEALTH_CHECKS_BY_OWNER: "MAX_HEALTH_CHECKS_BY_OWNER",
    MAX_HOSTED_ZONES_BY_OWNER: "MAX_HOSTED_ZONES_BY_OWNER",
    MAX_REUSABLE_DELEGATION_SETS_BY_OWNER: "MAX_REUSABLE_DELEGATION_SETS_BY_OWNER",
    MAX_TRAFFIC_POLICIES_BY_OWNER: "MAX_TRAFFIC_POLICIES_BY_OWNER",
    MAX_TRAFFIC_POLICY_INSTANCES_BY_OWNER: "MAX_TRAFFIC_POLICY_INSTANCES_BY_OWNER",
};
const ChangeStatus = {
    INSYNC: "INSYNC",
    PENDING: "PENDING",
};
let ConcurrentModification$1 = class ConcurrentModification extends Route53ServiceException$1 {
    name = "ConcurrentModification";
    $fault = "client";
    constructor(opts) {
        super({
            name: "ConcurrentModification",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ConcurrentModification.prototype);
    }
};
let InvalidInput$1 = class InvalidInput extends Route53ServiceException$1 {
    name = "InvalidInput";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidInput",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidInput.prototype);
    }
};
let InvalidKeySigningKeyStatus$1 = class InvalidKeySigningKeyStatus extends Route53ServiceException$1 {
    name = "InvalidKeySigningKeyStatus";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidKeySigningKeyStatus",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidKeySigningKeyStatus.prototype);
    }
};
let InvalidKMSArn$1 = class InvalidKMSArn extends Route53ServiceException$1 {
    name = "InvalidKMSArn";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidKMSArn",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidKMSArn.prototype);
    }
};
let InvalidSigningStatus$1 = class InvalidSigningStatus extends Route53ServiceException$1 {
    name = "InvalidSigningStatus";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidSigningStatus",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidSigningStatus.prototype);
    }
};
let NoSuchKeySigningKey$1 = class NoSuchKeySigningKey extends Route53ServiceException$1 {
    name = "NoSuchKeySigningKey";
    $fault = "client";
    constructor(opts) {
        super({
            name: "NoSuchKeySigningKey",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, NoSuchKeySigningKey.prototype);
    }
};
const CloudWatchRegion = {
    af_south_1: "af-south-1",
    ap_east_1: "ap-east-1",
    ap_east_2: "ap-east-2",
    ap_northeast_1: "ap-northeast-1",
    ap_northeast_2: "ap-northeast-2",
    ap_northeast_3: "ap-northeast-3",
    ap_south_1: "ap-south-1",
    ap_south_2: "ap-south-2",
    ap_southeast_1: "ap-southeast-1",
    ap_southeast_2: "ap-southeast-2",
    ap_southeast_3: "ap-southeast-3",
    ap_southeast_4: "ap-southeast-4",
    ap_southeast_5: "ap-southeast-5",
    ap_southeast_6: "ap-southeast-6",
    ap_southeast_7: "ap-southeast-7",
    ca_central_1: "ca-central-1",
    ca_west_1: "ca-west-1",
    cn_north_1: "cn-north-1",
    cn_northwest_1: "cn-northwest-1",
    eu_central_1: "eu-central-1",
    eu_central_2: "eu-central-2",
    eu_isoe_west_1: "eu-isoe-west-1",
    eu_north_1: "eu-north-1",
    eu_south_1: "eu-south-1",
    eu_south_2: "eu-south-2",
    eu_west_1: "eu-west-1",
    eu_west_2: "eu-west-2",
    eu_west_3: "eu-west-3",
    il_central_1: "il-central-1",
    me_central_1: "me-central-1",
    me_south_1: "me-south-1",
    mx_central_1: "mx-central-1",
    sa_east_1: "sa-east-1",
    us_east_1: "us-east-1",
    us_east_2: "us-east-2",
    us_gov_east_1: "us-gov-east-1",
    us_gov_west_1: "us-gov-west-1",
    us_iso_east_1: "us-iso-east-1",
    us_iso_west_1: "us-iso-west-1",
    us_isob_east_1: "us-isob-east-1",
    us_isob_west_1: "us-isob-west-1",
    us_isof_east_1: "us-isof-east-1",
    us_isof_south_1: "us-isof-south-1",
    us_west_1: "us-west-1",
    us_west_2: "us-west-2",
};
const VPCRegion = {
    af_south_1: "af-south-1",
    ap_east_1: "ap-east-1",
    ap_east_2: "ap-east-2",
    ap_northeast_1: "ap-northeast-1",
    ap_northeast_2: "ap-northeast-2",
    ap_northeast_3: "ap-northeast-3",
    ap_south_1: "ap-south-1",
    ap_south_2: "ap-south-2",
    ap_southeast_1: "ap-southeast-1",
    ap_southeast_2: "ap-southeast-2",
    ap_southeast_3: "ap-southeast-3",
    ap_southeast_4: "ap-southeast-4",
    ap_southeast_5: "ap-southeast-5",
    ap_southeast_6: "ap-southeast-6",
    ap_southeast_7: "ap-southeast-7",
    ca_central_1: "ca-central-1",
    ca_west_1: "ca-west-1",
    cn_north_1: "cn-north-1",
    cn_northwest_1: "cn-northwest-1",
    eu_central_1: "eu-central-1",
    eu_central_2: "eu-central-2",
    eu_isoe_west_1: "eu-isoe-west-1",
    eu_north_1: "eu-north-1",
    eu_south_1: "eu-south-1",
    eu_south_2: "eu-south-2",
    eu_west_1: "eu-west-1",
    eu_west_2: "eu-west-2",
    eu_west_3: "eu-west-3",
    il_central_1: "il-central-1",
    me_central_1: "me-central-1",
    me_south_1: "me-south-1",
    mx_central_1: "mx-central-1",
    sa_east_1: "sa-east-1",
    us_east_1: "us-east-1",
    us_east_2: "us-east-2",
    us_gov_east_1: "us-gov-east-1",
    us_gov_west_1: "us-gov-west-1",
    us_iso_east_1: "us-iso-east-1",
    us_iso_west_1: "us-iso-west-1",
    us_isob_east_1: "us-isob-east-1",
    us_isob_west_1: "us-isob-west-1",
    us_isof_east_1: "us-isof-east-1",
    us_isof_south_1: "us-isof-south-1",
    us_west_1: "us-west-1",
    us_west_2: "us-west-2",
};
let ConflictingDomainExists$1 = class ConflictingDomainExists extends Route53ServiceException$1 {
    name = "ConflictingDomainExists";
    $fault = "client";
    constructor(opts) {
        super({
            name: "ConflictingDomainExists",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ConflictingDomainExists.prototype);
    }
};
let InvalidVPCId$1 = class InvalidVPCId extends Route53ServiceException$1 {
    name = "InvalidVPCId";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidVPCId",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidVPCId.prototype);
    }
};
let LimitsExceeded$1 = class LimitsExceeded extends Route53ServiceException$1 {
    name = "LimitsExceeded";
    $fault = "client";
    constructor(opts) {
        super({
            name: "LimitsExceeded",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, LimitsExceeded.prototype);
    }
};
let NoSuchHostedZone$1 = class NoSuchHostedZone extends Route53ServiceException$1 {
    name = "NoSuchHostedZone";
    $fault = "client";
    constructor(opts) {
        super({
            name: "NoSuchHostedZone",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, NoSuchHostedZone.prototype);
    }
};
let NotAuthorizedException$1 = class NotAuthorizedException extends Route53ServiceException$1 {
    name = "NotAuthorizedException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "NotAuthorizedException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, NotAuthorizedException.prototype);
    }
};
let PriorRequestNotComplete$1 = class PriorRequestNotComplete extends Route53ServiceException$1 {
    name = "PriorRequestNotComplete";
    $fault = "client";
    constructor(opts) {
        super({
            name: "PriorRequestNotComplete",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, PriorRequestNotComplete.prototype);
    }
};
let PublicZoneVPCAssociation$1 = class PublicZoneVPCAssociation extends Route53ServiceException$1 {
    name = "PublicZoneVPCAssociation";
    $fault = "client";
    constructor(opts) {
        super({
            name: "PublicZoneVPCAssociation",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, PublicZoneVPCAssociation.prototype);
    }
};
const CidrCollectionChangeAction = {
    DELETE_IF_EXISTS: "DELETE_IF_EXISTS",
    PUT: "PUT",
};
let CidrBlockInUseException$1 = class CidrBlockInUseException extends Route53ServiceException$1 {
    name = "CidrBlockInUseException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "CidrBlockInUseException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, CidrBlockInUseException.prototype);
        this.Message = opts.Message;
    }
};
let CidrCollectionVersionMismatchException$1 = class CidrCollectionVersionMismatchException extends Route53ServiceException$1 {
    name = "CidrCollectionVersionMismatchException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "CidrCollectionVersionMismatchException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, CidrCollectionVersionMismatchException.prototype);
        this.Message = opts.Message;
    }
};
let NoSuchCidrCollectionException$1 = class NoSuchCidrCollectionException extends Route53ServiceException$1 {
    name = "NoSuchCidrCollectionException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "NoSuchCidrCollectionException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, NoSuchCidrCollectionException.prototype);
        this.Message = opts.Message;
    }
};
const ChangeAction = {
    CREATE: "CREATE",
    DELETE: "DELETE",
    UPSERT: "UPSERT",
};
const ResourceRecordSetFailover = {
    PRIMARY: "PRIMARY",
    SECONDARY: "SECONDARY",
};
const ResourceRecordSetRegion = {
    af_south_1: "af-south-1",
    ap_east_1: "ap-east-1",
    ap_east_2: "ap-east-2",
    ap_northeast_1: "ap-northeast-1",
    ap_northeast_2: "ap-northeast-2",
    ap_northeast_3: "ap-northeast-3",
    ap_south_1: "ap-south-1",
    ap_south_2: "ap-south-2",
    ap_southeast_1: "ap-southeast-1",
    ap_southeast_2: "ap-southeast-2",
    ap_southeast_3: "ap-southeast-3",
    ap_southeast_4: "ap-southeast-4",
    ap_southeast_5: "ap-southeast-5",
    ap_southeast_6: "ap-southeast-6",
    ap_southeast_7: "ap-southeast-7",
    ca_central_1: "ca-central-1",
    ca_west_1: "ca-west-1",
    cn_north_1: "cn-north-1",
    cn_northwest_1: "cn-northwest-1",
    eu_central_1: "eu-central-1",
    eu_central_2: "eu-central-2",
    eu_north_1: "eu-north-1",
    eu_south_1: "eu-south-1",
    eu_south_2: "eu-south-2",
    eu_west_1: "eu-west-1",
    eu_west_2: "eu-west-2",
    eu_west_3: "eu-west-3",
    il_central_1: "il-central-1",
    me_central_1: "me-central-1",
    me_south_1: "me-south-1",
    mx_central_1: "mx-central-1",
    sa_east_1: "sa-east-1",
    us_east_1: "us-east-1",
    us_east_2: "us-east-2",
    us_gov_east_1: "us-gov-east-1",
    us_gov_west_1: "us-gov-west-1",
    us_west_1: "us-west-1",
    us_west_2: "us-west-2",
};
const RRType = {
    A: "A",
    AAAA: "AAAA",
    CAA: "CAA",
    CNAME: "CNAME",
    DS: "DS",
    HTTPS: "HTTPS",
    MX: "MX",
    NAPTR: "NAPTR",
    NS: "NS",
    PTR: "PTR",
    SOA: "SOA",
    SPF: "SPF",
    SRV: "SRV",
    SSHFP: "SSHFP",
    SVCB: "SVCB",
    TLSA: "TLSA",
    TXT: "TXT",
};
let InvalidChangeBatch$1 = class InvalidChangeBatch extends Route53ServiceException$1 {
    name = "InvalidChangeBatch";
    $fault = "client";
    messages;
    constructor(opts) {
        super({
            name: "InvalidChangeBatch",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidChangeBatch.prototype);
        this.messages = opts.messages;
    }
};
let NoSuchHealthCheck$1 = class NoSuchHealthCheck extends Route53ServiceException$1 {
    name = "NoSuchHealthCheck";
    $fault = "client";
    constructor(opts) {
        super({
            name: "NoSuchHealthCheck",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, NoSuchHealthCheck.prototype);
    }
};
const TagResourceType = {
    healthcheck: "healthcheck",
    hostedzone: "hostedzone",
};
let ThrottlingException$1 = class ThrottlingException extends Route53ServiceException$1 {
    name = "ThrottlingException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "ThrottlingException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ThrottlingException.prototype);
    }
};
let CidrCollectionAlreadyExistsException$1 = class CidrCollectionAlreadyExistsException extends Route53ServiceException$1 {
    name = "CidrCollectionAlreadyExistsException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "CidrCollectionAlreadyExistsException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, CidrCollectionAlreadyExistsException.prototype);
        this.Message = opts.Message;
    }
};
const InsufficientDataHealthStatus = {
    Healthy: "Healthy",
    LastKnownStatus: "LastKnownStatus",
    Unhealthy: "Unhealthy",
};
const HealthCheckRegion = {
    ap_northeast_1: "ap-northeast-1",
    ap_southeast_1: "ap-southeast-1",
    ap_southeast_2: "ap-southeast-2",
    eu_west_1: "eu-west-1",
    sa_east_1: "sa-east-1",
    us_east_1: "us-east-1",
    us_west_1: "us-west-1",
    us_west_2: "us-west-2",
};
const HealthCheckType = {
    CALCULATED: "CALCULATED",
    CLOUDWATCH_METRIC: "CLOUDWATCH_METRIC",
    HTTP: "HTTP",
    HTTPS: "HTTPS",
    HTTPS_STR_MATCH: "HTTPS_STR_MATCH",
    HTTP_STR_MATCH: "HTTP_STR_MATCH",
    RECOVERY_CONTROL: "RECOVERY_CONTROL",
    TCP: "TCP",
};
const ComparisonOperator = {
    GreaterThanOrEqualToThreshold: "GreaterThanOrEqualToThreshold",
    GreaterThanThreshold: "GreaterThanThreshold",
    LessThanOrEqualToThreshold: "LessThanOrEqualToThreshold",
    LessThanThreshold: "LessThanThreshold",
};
const Statistic = {
    Average: "Average",
    Maximum: "Maximum",
    Minimum: "Minimum",
    SampleCount: "SampleCount",
    Sum: "Sum",
};
let HealthCheckAlreadyExists$1 = class HealthCheckAlreadyExists extends Route53ServiceException$1 {
    name = "HealthCheckAlreadyExists";
    $fault = "client";
    constructor(opts) {
        super({
            name: "HealthCheckAlreadyExists",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, HealthCheckAlreadyExists.prototype);
    }
};
let TooManyHealthChecks$1 = class TooManyHealthChecks extends Route53ServiceException$1 {
    name = "TooManyHealthChecks";
    $fault = "client";
    constructor(opts) {
        super({
            name: "TooManyHealthChecks",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TooManyHealthChecks.prototype);
    }
};
let DelegationSetNotAvailable$1 = class DelegationSetNotAvailable extends Route53ServiceException$1 {
    name = "DelegationSetNotAvailable";
    $fault = "client";
    constructor(opts) {
        super({
            name: "DelegationSetNotAvailable",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, DelegationSetNotAvailable.prototype);
    }
};
let DelegationSetNotReusable$1 = class DelegationSetNotReusable extends Route53ServiceException$1 {
    name = "DelegationSetNotReusable";
    $fault = "client";
    constructor(opts) {
        super({
            name: "DelegationSetNotReusable",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, DelegationSetNotReusable.prototype);
    }
};
let HostedZoneAlreadyExists$1 = class HostedZoneAlreadyExists extends Route53ServiceException$1 {
    name = "HostedZoneAlreadyExists";
    $fault = "client";
    constructor(opts) {
        super({
            name: "HostedZoneAlreadyExists",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, HostedZoneAlreadyExists.prototype);
    }
};
let InvalidDomainName$1 = class InvalidDomainName extends Route53ServiceException$1 {
    name = "InvalidDomainName";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidDomainName",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidDomainName.prototype);
    }
};
let NoSuchDelegationSet$1 = class NoSuchDelegationSet extends Route53ServiceException$1 {
    name = "NoSuchDelegationSet";
    $fault = "client";
    constructor(opts) {
        super({
            name: "NoSuchDelegationSet",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, NoSuchDelegationSet.prototype);
    }
};
let TooManyHostedZones$1 = class TooManyHostedZones extends Route53ServiceException$1 {
    name = "TooManyHostedZones";
    $fault = "client";
    constructor(opts) {
        super({
            name: "TooManyHostedZones",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TooManyHostedZones.prototype);
    }
};
let InvalidArgument$1 = class InvalidArgument extends Route53ServiceException$1 {
    name = "InvalidArgument";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidArgument",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidArgument.prototype);
    }
};
let InvalidKeySigningKeyName$1 = class InvalidKeySigningKeyName extends Route53ServiceException$1 {
    name = "InvalidKeySigningKeyName";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidKeySigningKeyName",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidKeySigningKeyName.prototype);
    }
};
let KeySigningKeyAlreadyExists$1 = class KeySigningKeyAlreadyExists extends Route53ServiceException$1 {
    name = "KeySigningKeyAlreadyExists";
    $fault = "client";
    constructor(opts) {
        super({
            name: "KeySigningKeyAlreadyExists",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, KeySigningKeyAlreadyExists.prototype);
    }
};
let TooManyKeySigningKeys$1 = class TooManyKeySigningKeys extends Route53ServiceException$1 {
    name = "TooManyKeySigningKeys";
    $fault = "client";
    constructor(opts) {
        super({
            name: "TooManyKeySigningKeys",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TooManyKeySigningKeys.prototype);
    }
};
let InsufficientCloudWatchLogsResourcePolicy$1 = class InsufficientCloudWatchLogsResourcePolicy extends Route53ServiceException$1 {
    name = "InsufficientCloudWatchLogsResourcePolicy";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InsufficientCloudWatchLogsResourcePolicy",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InsufficientCloudWatchLogsResourcePolicy.prototype);
    }
};
let NoSuchCloudWatchLogsLogGroup$1 = class NoSuchCloudWatchLogsLogGroup extends Route53ServiceException$1 {
    name = "NoSuchCloudWatchLogsLogGroup";
    $fault = "client";
    constructor(opts) {
        super({
            name: "NoSuchCloudWatchLogsLogGroup",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, NoSuchCloudWatchLogsLogGroup.prototype);
    }
};
let QueryLoggingConfigAlreadyExists$1 = class QueryLoggingConfigAlreadyExists extends Route53ServiceException$1 {
    name = "QueryLoggingConfigAlreadyExists";
    $fault = "client";
    constructor(opts) {
        super({
            name: "QueryLoggingConfigAlreadyExists",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, QueryLoggingConfigAlreadyExists.prototype);
    }
};
let DelegationSetAlreadyCreated$1 = class DelegationSetAlreadyCreated extends Route53ServiceException$1 {
    name = "DelegationSetAlreadyCreated";
    $fault = "client";
    constructor(opts) {
        super({
            name: "DelegationSetAlreadyCreated",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, DelegationSetAlreadyCreated.prototype);
    }
};
let DelegationSetAlreadyReusable$1 = class DelegationSetAlreadyReusable extends Route53ServiceException$1 {
    name = "DelegationSetAlreadyReusable";
    $fault = "client";
    constructor(opts) {
        super({
            name: "DelegationSetAlreadyReusable",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, DelegationSetAlreadyReusable.prototype);
    }
};
let HostedZoneNotFound$1 = class HostedZoneNotFound extends Route53ServiceException$1 {
    name = "HostedZoneNotFound";
    $fault = "client";
    constructor(opts) {
        super({
            name: "HostedZoneNotFound",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, HostedZoneNotFound.prototype);
    }
};
let InvalidTrafficPolicyDocument$1 = class InvalidTrafficPolicyDocument extends Route53ServiceException$1 {
    name = "InvalidTrafficPolicyDocument";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidTrafficPolicyDocument",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidTrafficPolicyDocument.prototype);
    }
};
let TooManyTrafficPolicies$1 = class TooManyTrafficPolicies extends Route53ServiceException$1 {
    name = "TooManyTrafficPolicies";
    $fault = "client";
    constructor(opts) {
        super({
            name: "TooManyTrafficPolicies",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TooManyTrafficPolicies.prototype);
    }
};
let TrafficPolicyAlreadyExists$1 = class TrafficPolicyAlreadyExists extends Route53ServiceException$1 {
    name = "TrafficPolicyAlreadyExists";
    $fault = "client";
    constructor(opts) {
        super({
            name: "TrafficPolicyAlreadyExists",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TrafficPolicyAlreadyExists.prototype);
    }
};
let NoSuchTrafficPolicy$1 = class NoSuchTrafficPolicy extends Route53ServiceException$1 {
    name = "NoSuchTrafficPolicy";
    $fault = "client";
    constructor(opts) {
        super({
            name: "NoSuchTrafficPolicy",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, NoSuchTrafficPolicy.prototype);
    }
};
let TooManyTrafficPolicyInstances$1 = class TooManyTrafficPolicyInstances extends Route53ServiceException$1 {
    name = "TooManyTrafficPolicyInstances";
    $fault = "client";
    constructor(opts) {
        super({
            name: "TooManyTrafficPolicyInstances",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TooManyTrafficPolicyInstances.prototype);
    }
};
let TrafficPolicyInstanceAlreadyExists$1 = class TrafficPolicyInstanceAlreadyExists extends Route53ServiceException$1 {
    name = "TrafficPolicyInstanceAlreadyExists";
    $fault = "client";
    constructor(opts) {
        super({
            name: "TrafficPolicyInstanceAlreadyExists",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TrafficPolicyInstanceAlreadyExists.prototype);
    }
};
let TooManyTrafficPolicyVersionsForCurrentPolicy$1 = class TooManyTrafficPolicyVersionsForCurrentPolicy extends Route53ServiceException$1 {
    name = "TooManyTrafficPolicyVersionsForCurrentPolicy";
    $fault = "client";
    constructor(opts) {
        super({
            name: "TooManyTrafficPolicyVersionsForCurrentPolicy",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TooManyTrafficPolicyVersionsForCurrentPolicy.prototype);
    }
};
let TooManyVPCAssociationAuthorizations$1 = class TooManyVPCAssociationAuthorizations extends Route53ServiceException$1 {
    name = "TooManyVPCAssociationAuthorizations";
    $fault = "client";
    constructor(opts) {
        super({
            name: "TooManyVPCAssociationAuthorizations",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TooManyVPCAssociationAuthorizations.prototype);
    }
};
let KeySigningKeyInParentDSRecord$1 = class KeySigningKeyInParentDSRecord extends Route53ServiceException$1 {
    name = "KeySigningKeyInParentDSRecord";
    $fault = "client";
    constructor(opts) {
        super({
            name: "KeySigningKeyInParentDSRecord",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, KeySigningKeyInParentDSRecord.prototype);
    }
};
let KeySigningKeyInUse$1 = class KeySigningKeyInUse extends Route53ServiceException$1 {
    name = "KeySigningKeyInUse";
    $fault = "client";
    constructor(opts) {
        super({
            name: "KeySigningKeyInUse",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, KeySigningKeyInUse.prototype);
    }
};
let CidrCollectionInUseException$1 = class CidrCollectionInUseException extends Route53ServiceException$1 {
    name = "CidrCollectionInUseException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "CidrCollectionInUseException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, CidrCollectionInUseException.prototype);
        this.Message = opts.Message;
    }
};
let HealthCheckInUse$1 = class HealthCheckInUse extends Route53ServiceException$1 {
    name = "HealthCheckInUse";
    $fault = "client";
    constructor(opts) {
        super({
            name: "HealthCheckInUse",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, HealthCheckInUse.prototype);
    }
};
let HostedZoneNotEmpty$1 = class HostedZoneNotEmpty extends Route53ServiceException$1 {
    name = "HostedZoneNotEmpty";
    $fault = "client";
    constructor(opts) {
        super({
            name: "HostedZoneNotEmpty",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, HostedZoneNotEmpty.prototype);
    }
};
let NoSuchQueryLoggingConfig$1 = class NoSuchQueryLoggingConfig extends Route53ServiceException$1 {
    name = "NoSuchQueryLoggingConfig";
    $fault = "client";
    constructor(opts) {
        super({
            name: "NoSuchQueryLoggingConfig",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, NoSuchQueryLoggingConfig.prototype);
    }
};
let DelegationSetInUse$1 = class DelegationSetInUse extends Route53ServiceException$1 {
    name = "DelegationSetInUse";
    $fault = "client";
    constructor(opts) {
        super({
            name: "DelegationSetInUse",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, DelegationSetInUse.prototype);
    }
};
let TrafficPolicyInUse$1 = class TrafficPolicyInUse extends Route53ServiceException$1 {
    name = "TrafficPolicyInUse";
    $fault = "client";
    constructor(opts) {
        super({
            name: "TrafficPolicyInUse",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TrafficPolicyInUse.prototype);
    }
};
let NoSuchTrafficPolicyInstance$1 = class NoSuchTrafficPolicyInstance extends Route53ServiceException$1 {
    name = "NoSuchTrafficPolicyInstance";
    $fault = "client";
    constructor(opts) {
        super({
            name: "NoSuchTrafficPolicyInstance",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, NoSuchTrafficPolicyInstance.prototype);
    }
};
let VPCAssociationAuthorizationNotFound$1 = class VPCAssociationAuthorizationNotFound extends Route53ServiceException$1 {
    name = "VPCAssociationAuthorizationNotFound";
    $fault = "client";
    constructor(opts) {
        super({
            name: "VPCAssociationAuthorizationNotFound",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, VPCAssociationAuthorizationNotFound.prototype);
    }
};
let DNSSECNotFound$1 = class DNSSECNotFound extends Route53ServiceException$1 {
    name = "DNSSECNotFound";
    $fault = "client";
    constructor(opts) {
        super({
            name: "DNSSECNotFound",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, DNSSECNotFound.prototype);
    }
};
let LastVPCAssociation$1 = class LastVPCAssociation extends Route53ServiceException$1 {
    name = "LastVPCAssociation";
    $fault = "client";
    constructor(opts) {
        super({
            name: "LastVPCAssociation",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, LastVPCAssociation.prototype);
    }
};
let VPCAssociationNotFound$1 = class VPCAssociationNotFound extends Route53ServiceException$1 {
    name = "VPCAssociationNotFound";
    $fault = "client";
    constructor(opts) {
        super({
            name: "VPCAssociationNotFound",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, VPCAssociationNotFound.prototype);
    }
};
let HostedZonePartiallyDelegated$1 = class HostedZonePartiallyDelegated extends Route53ServiceException$1 {
    name = "HostedZonePartiallyDelegated";
    $fault = "client";
    constructor(opts) {
        super({
            name: "HostedZonePartiallyDelegated",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, HostedZonePartiallyDelegated.prototype);
    }
};
let KeySigningKeyWithActiveStatusNotFound$1 = class KeySigningKeyWithActiveStatusNotFound extends Route53ServiceException$1 {
    name = "KeySigningKeyWithActiveStatusNotFound";
    $fault = "client";
    constructor(opts) {
        super({
            name: "KeySigningKeyWithActiveStatusNotFound",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, KeySigningKeyWithActiveStatusNotFound.prototype);
    }
};
let NoSuchChange$1 = class NoSuchChange extends Route53ServiceException$1 {
    name = "NoSuchChange";
    $fault = "client";
    constructor(opts) {
        super({
            name: "NoSuchChange",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, NoSuchChange.prototype);
    }
};
let NoSuchGeoLocation$1 = class NoSuchGeoLocation extends Route53ServiceException$1 {
    name = "NoSuchGeoLocation";
    $fault = "client";
    constructor(opts) {
        super({
            name: "NoSuchGeoLocation",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, NoSuchGeoLocation.prototype);
    }
};
let IncompatibleVersion$1 = class IncompatibleVersion extends Route53ServiceException$1 {
    name = "IncompatibleVersion";
    $fault = "client";
    constructor(opts) {
        super({
            name: "IncompatibleVersion",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, IncompatibleVersion.prototype);
    }
};
const HostedZoneLimitType = {
    MAX_RRSETS_BY_ZONE: "MAX_RRSETS_BY_ZONE",
    MAX_VPCS_ASSOCIATED_BY_ZONE: "MAX_VPCS_ASSOCIATED_BY_ZONE",
};
let HostedZoneNotPrivate$1 = class HostedZoneNotPrivate extends Route53ServiceException$1 {
    name = "HostedZoneNotPrivate";
    $fault = "client";
    constructor(opts) {
        super({
            name: "HostedZoneNotPrivate",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, HostedZoneNotPrivate.prototype);
    }
};
const ReusableDelegationSetLimitType = {
    MAX_ZONES_BY_REUSABLE_DELEGATION_SET: "MAX_ZONES_BY_REUSABLE_DELEGATION_SET",
};
let NoSuchCidrLocationException$1 = class NoSuchCidrLocationException extends Route53ServiceException$1 {
    name = "NoSuchCidrLocationException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "NoSuchCidrLocationException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, NoSuchCidrLocationException.prototype);
        this.Message = opts.Message;
    }
};
const HostedZoneType = {
    PRIVATE_HOSTED_ZONE: "PrivateHostedZone",
};
let InvalidPaginationToken$1 = class InvalidPaginationToken extends Route53ServiceException$1 {
    name = "InvalidPaginationToken";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidPaginationToken",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidPaginationToken.prototype);
    }
};
let HealthCheckVersionMismatch$1 = class HealthCheckVersionMismatch extends Route53ServiceException$1 {
    name = "HealthCheckVersionMismatch";
    $fault = "client";
    constructor(opts) {
        super({
            name: "HealthCheckVersionMismatch",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, HealthCheckVersionMismatch.prototype);
    }
};
const ResettableElementName = {
    ChildHealthChecks: "ChildHealthChecks",
    FullyQualifiedDomainName: "FullyQualifiedDomainName",
    Regions: "Regions",
    ResourcePath: "ResourcePath",
};
let ConflictingTypes$1 = class ConflictingTypes extends Route53ServiceException$1 {
    name = "ConflictingTypes";
    $fault = "client";
    constructor(opts) {
        super({
            name: "ConflictingTypes",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ConflictingTypes.prototype);
    }
};

const _A = "Action";
const _AI = "AlarmIdentifier";
const _AKSK = "ActivateKeySigningKey";
const _AKSKR = "ActivateKeySigningKeyRequest";
const _AKSKRc = "ActivateKeySigningKeyResponse";
const _AL = "AccountLimit";
const _AT = "AliasTarget";
const _ATd = "AddTags";
const _AVPCWHZ = "AssociateVPCWithHostedZone";
const _AVPCWHZR = "AssociateVPCWithHostedZoneRequest";
const _AVPCWHZRs = "AssociateVPCWithHostedZoneResponse";
const _AWSR = "AWSRegion";
const _Ar = "Arn";
const _B = "Bias";
const _C = "Comment";
const _CB = "ChangeBatch";
const _CBIUE = "CidrBlockInUseException";
const _CBS = "CidrBlockSummary";
const _CBSi = "CidrBlockSummaries";
const _CBi = "CidrBlock";
const _CBid = "CidrBlocks";
const _CC = "CidrCollection";
const _CCAEE = "CidrCollectionAlreadyExistsException";
const _CCC = "CidrCollectionChange";
const _CCCR = "ChangeCidrCollectionRequest";
const _CCCRh = "ChangeCidrCollectionResponse";
const _CCCRr = "CreateCidrCollectionRequest";
const _CCCRre = "CreateCidrCollectionResponse";
const _CCCh = "ChangeCidrCollection";
const _CCCi = "CidrCollectionChanges";
const _CCCr = "CreateCidrCollection";
const _CCIUE = "CidrCollectionInUseException";
const _CCVME = "CidrCollectionVersionMismatchException";
const _CCi = "CidrCollections";
const _CCo = "ContinentCode";
const _CCou = "CountryCode";
const _CD = "CreatedDate";
const _CDE = "ConflictingDomainExists";
const _CHC = "ChildHealthChecks";
const _CHCL = "ChildHealthCheckList";
const _CHCR = "CreateHealthCheckRequest";
const _CHCRr = "CreateHealthCheckResponse";
const _CHCh = "ChildHealthCheck";
const _CHCr = "CreateHealthCheck";
const _CHZ = "CreateHostedZone";
const _CHZR = "CreateHostedZoneRequest";
const _CHZRr = "CreateHostedZoneResponse";
const _CI = "ChangeInfo";
const _CIR = "CheckerIpRanges";
const _CIo = "CollectionId";
const _CKSK = "CreateKeySigningKey";
const _CKSKR = "CreateKeySigningKeyRequest";
const _CKSKRr = "CreateKeySigningKeyResponse";
const _CL = "CidrList";
const _CLi = "CidrLocations";
const _CM = "ConcurrentModification";
const _CN = "ContinentName";
const _CNo = "CountryName";
const _CO = "ComparisonOperator";
const _CQLC = "CreateQueryLoggingConfig";
const _CQLCR = "CreateQueryLoggingConfigRequest";
const _CQLCRr = "CreateQueryLoggingConfigResponse";
const _CR = "CallerReference";
const _CRC = "CidrRoutingConfig";
const _CRDS = "CreateReusableDelegationSet";
const _CRDSR = "CreateReusableDelegationSetRequest";
const _CRDSRr = "CreateReusableDelegationSetResponse";
const _CRRS = "ChangeResourceRecordSets";
const _CRRSR = "ChangeResourceRecordSetsRequest";
const _CRRSRh = "ChangeResourceRecordSetsResponse";
const _CS = "CollectionSummary";
const _CSo = "CollectionSummaries";
const _CT = "ConflictingTypes";
const _CTFR = "ChangeTagsForResource";
const _CTFRR = "ChangeTagsForResourceRequest";
const _CTFRRh = "ChangeTagsForResourceResponse";
const _CTP = "CreateTrafficPolicy";
const _CTPI = "CreateTrafficPolicyInstance";
const _CTPIR = "CreateTrafficPolicyInstanceRequest";
const _CTPIRr = "CreateTrafficPolicyInstanceResponse";
const _CTPR = "CreateTrafficPolicyRequest";
const _CTPRr = "CreateTrafficPolicyResponse";
const _CTPV = "CreateTrafficPolicyVersion";
const _CTPVR = "CreateTrafficPolicyVersionRequest";
const _CTPVRr = "CreateTrafficPolicyVersionResponse";
const _CTh = "CheckedTime";
const _CV = "CollectionVersion";
const _CVPCAA = "CreateVPCAssociationAuthorization";
const _CVPCAAR = "CreateVPCAssociationAuthorizationRequest";
const _CVPCAARr = "CreateVPCAssociationAuthorizationResponse";
const _CWAC = "CloudWatchAlarmConfiguration";
const _CWLLGA = "CloudWatchLogsLogGroupArn";
const _Ch = "Change";
const _Cha = "Changes";
const _Ci = "Cidr";
const _Co = "Coordinates";
const _Col = "Collection";
const _Con = "Config";
const _Cou = "Count";
const _D = "Dimensions";
const _DAM = "DigestAlgorithmMnemonic";
const _DAT = "DigestAlgorithmType";
const _DCC = "DeleteCidrCollection";
const _DCCR = "DeleteCidrCollectionRequest";
const _DCCRe = "DeleteCidrCollectionResponse";
const _DHC = "DeleteHealthCheck";
const _DHCR = "DeleteHealthCheckRequest";
const _DHCRe = "DeleteHealthCheckResponse";
const _DHZ = "DeleteHostedZone";
const _DHZDNSSEC = "DisableHostedZoneDNSSEC";
const _DHZDNSSECR = "DisableHostedZoneDNSSECRequest";
const _DHZDNSSECRi = "DisableHostedZoneDNSSECResponse";
const _DHZR = "DeleteHostedZoneRequest";
const _DHZRe = "DeleteHostedZoneResponse";
const _DKSK = "DeactivateKeySigningKey";
const _DKSKR = "DeactivateKeySigningKeyRequest";
const _DKSKRe = "DeactivateKeySigningKeyResponse";
const _DKSKRel = "DeleteKeySigningKeyRequest";
const _DKSKRele = "DeleteKeySigningKeyResponse";
const _DKSKe = "DeleteKeySigningKey";
const _DL = "DimensionList";
const _DNSKEYR = "DNSKEYRecord";
const _DNSN = "DNSName";
const _DNSSECNF = "DNSSECNotFound";
const _DNSSECS = "DNSSECStatus";
const _DQLC = "DeleteQueryLoggingConfig";
const _DQLCR = "DeleteQueryLoggingConfigRequest";
const _DQLCRe = "DeleteQueryLoggingConfigResponse";
const _DRDS = "DeleteReusableDelegationSet";
const _DRDSR = "DeleteReusableDelegationSetRequest";
const _DRDSRe = "DeleteReusableDelegationSetResponse";
const _DS = "DelegationSet";
const _DSAC = "DelegationSetAlreadyCreated";
const _DSAR = "DelegationSetAlreadyReusable";
const _DSI = "DelegationSetId";
const _DSIU = "DelegationSetInUse";
const _DSNA = "DelegationSetNotAvailable";
const _DSNR = "DelegationSetNotReusable";
const _DSNS = "DelegationSetNameServers";
const _DSR = "DSRecord";
const _DSe = "DelegationSets";
const _DTP = "DeleteTrafficPolicy";
const _DTPI = "DeleteTrafficPolicyInstance";
const _DTPIR = "DeleteTrafficPolicyInstanceRequest";
const _DTPIRe = "DeleteTrafficPolicyInstanceResponse";
const _DTPR = "DeleteTrafficPolicyRequest";
const _DTPRe = "DeleteTrafficPolicyResponse";
const _DV = "DigestValue";
const _DVPCAA = "DeleteVPCAssociationAuthorization";
const _DVPCAAR = "DeleteVPCAssociationAuthorizationRequest";
const _DVPCAARe = "DeleteVPCAssociationAuthorizationResponse";
const _DVPCFHZ = "DisassociateVPCFromHostedZone";
const _DVPCFHZR = "DisassociateVPCFromHostedZoneRequest";
const _DVPCFHZRi = "DisassociateVPCFromHostedZoneResponse";
const _De = "Description";
const _Di = "Dimension";
const _Dis = "Disabled";
const _Do = "Document";
const _EDNSCSIP = "EDNS0ClientSubnetIP";
const _EDNSCSM = "EDNS0ClientSubnetMask";
const _EHZDNSSEC = "EnableHostedZoneDNSSEC";
const _EHZDNSSECR = "EnableHostedZoneDNSSECRequest";
const _EHZDNSSECRn = "EnableHostedZoneDNSSECResponse";
const _EM = "ErrorMessages";
const _EP = "EvaluationPeriods";
const _ESNI = "EnableSNI";
const _ETH = "EvaluateTargetHealth";
const _F = "Flag";
const _FQDN = "FullyQualifiedDomainName";
const _FT = "FailureThreshold";
const _Fa = "Failover";
const _GAL = "GetAccountLimit";
const _GALR = "GetAccountLimitRequest";
const _GALRe = "GetAccountLimitResponse";
const _GC = "GetChange";
const _GCIR = "GetCheckerIpRanges";
const _GCIRR = "GetCheckerIpRangesRequest";
const _GCIRRe = "GetCheckerIpRangesResponse";
const _GCR = "GetChangeRequest";
const _GCRe = "GetChangeResponse";
const _GDNSSEC = "GetDNSSEC";
const _GDNSSECR = "GetDNSSECRequest";
const _GDNSSECRe = "GetDNSSECResponse";
const _GGL = "GetGeoLocation";
const _GGLR = "GetGeoLocationRequest";
const _GGLRe = "GetGeoLocationResponse";
const _GHC = "GetHealthCheck";
const _GHCC = "GetHealthCheckCount";
const _GHCCR = "GetHealthCheckCountRequest";
const _GHCCRe = "GetHealthCheckCountResponse";
const _GHCLFR = "GetHealthCheckLastFailureReason";
const _GHCLFRR = "GetHealthCheckLastFailureReasonRequest";
const _GHCLFRRe = "GetHealthCheckLastFailureReasonResponse";
const _GHCR = "GetHealthCheckRequest";
const _GHCRe = "GetHealthCheckResponse";
const _GHCS = "GetHealthCheckStatus";
const _GHCSR = "GetHealthCheckStatusRequest";
const _GHCSRe = "GetHealthCheckStatusResponse";
const _GHZ = "GetHostedZone";
const _GHZC = "GetHostedZoneCount";
const _GHZCR = "GetHostedZoneCountRequest";
const _GHZCRe = "GetHostedZoneCountResponse";
const _GHZL = "GetHostedZoneLimit";
const _GHZLR = "GetHostedZoneLimitRequest";
const _GHZLRe = "GetHostedZoneLimitResponse";
const _GHZR = "GetHostedZoneRequest";
const _GHZRe = "GetHostedZoneResponse";
const _GL = "GeoLocation";
const _GLD = "GeoLocationDetails";
const _GLDL = "GeoLocationDetailsList";
const _GPL = "GeoProximityLocation";
const _GQLC = "GetQueryLoggingConfig";
const _GQLCR = "GetQueryLoggingConfigRequest";
const _GQLCRe = "GetQueryLoggingConfigResponse";
const _GRDS = "GetReusableDelegationSet";
const _GRDSL = "GetReusableDelegationSetLimit";
const _GRDSLR = "GetReusableDelegationSetLimitRequest";
const _GRDSLRe = "GetReusableDelegationSetLimitResponse";
const _GRDSR = "GetReusableDelegationSetRequest";
const _GRDSRe = "GetReusableDelegationSetResponse";
const _GTP = "GetTrafficPolicy";
const _GTPI = "GetTrafficPolicyInstance";
const _GTPIC = "GetTrafficPolicyInstanceCount";
const _GTPICR = "GetTrafficPolicyInstanceCountRequest";
const _GTPICRe = "GetTrafficPolicyInstanceCountResponse";
const _GTPIR = "GetTrafficPolicyInstanceRequest";
const _GTPIRe = "GetTrafficPolicyInstanceResponse";
const _GTPR = "GetTrafficPolicyRequest";
const _GTPRe = "GetTrafficPolicyResponse";
const _HC = "HealthCheck";
const _HCAE = "HealthCheckAlreadyExists";
const _HCC = "HealthCheckConfig";
const _HCCe = "HealthCheckCount";
const _HCI = "HealthCheckId";
const _HCIU = "HealthCheckInUse";
const _HCO = "HealthCheckObservations";
const _HCOe = "HealthCheckObservation";
const _HCRL = "HealthCheckRegionList";
const _HCV = "HealthCheckVersion";
const _HCVM = "HealthCheckVersionMismatch";
const _HCe = "HealthChecks";
const _HT = "HealthThreshold";
const _HZ = "HostedZone";
const _HZAE = "HostedZoneAlreadyExists";
const _HZC = "HostedZoneConfig";
const _HZCo = "HostedZoneCount";
const _HZI = "HostedZoneId";
const _HZIM = "HostedZoneIdMarker";
const _HZL = "HostedZoneLimit";
const _HZNE = "HostedZoneNotEmpty";
const _HZNF = "HostedZoneNotFound";
const _HZNP = "HostedZoneNotPrivate";
const _HZO = "HostedZoneOwner";
const _HZPD = "HostedZonePartiallyDelegated";
const _HZS = "HostedZoneSummary";
const _HZSo = "HostedZoneSummaries";
const _HZT = "HostedZoneType";
const _HZo = "HostedZones";
const _I = "Id";
const _IA = "InvalidArgument";
const _ICB = "InvalidChangeBatch";
const _ICWLRP = "InsufficientCloudWatchLogsResourcePolicy";
const _IDHS = "InsufficientDataHealthStatus";
const _IDN = "InvalidDomainName";
const _II = "InvalidInput";
const _IKMSA = "InvalidKMSArn";
const _IKSKN = "InvalidKeySigningKeyName";
const _IKSKS = "InvalidKeySigningKeyStatus";
const _IPA = "IPAddress";
const _IPT = "InvalidPaginationToken";
const _ISS = "InvalidSigningStatus";
const _IT = "IsTruncated";
const _ITPD = "InvalidTrafficPolicyDocument";
const _IV = "IncompatibleVersion";
const _IVPCI = "InvalidVPCId";
const _In = "Inverted";
const _K = "Key";
const _KA = "KmsArn";
const _KMSA = "KeyManagementServiceArn";
const _KSK = "KeySigningKey";
const _KSKAE = "KeySigningKeyAlreadyExists";
const _KSKIPDSR = "KeySigningKeyInParentDSRecord";
const _KSKIU = "KeySigningKeyInUse";
const _KSKWASNF = "KeySigningKeyWithActiveStatusNotFound";
const _KSKe = "KeySigningKeys";
const _KT = "KeyTag";
const _L = "Latitude";
const _LCB = "ListCidrBlocks";
const _LCBR = "ListCidrBlocksRequest";
const _LCBRi = "ListCidrBlocksResponse";
const _LCC = "ListCidrCollections";
const _LCCR = "ListCidrCollectionsRequest";
const _LCCRi = "ListCidrCollectionsResponse";
const _LCL = "ListCidrLocations";
const _LCLR = "ListCidrLocationsRequest";
const _LCLRi = "ListCidrLocationsResponse";
const _LE = "LimitsExceeded";
const _LGL = "ListGeoLocations";
const _LGLR = "ListGeoLocationsRequest";
const _LGLRi = "ListGeoLocationsResponse";
const _LHC = "ListHealthChecks";
const _LHCR = "ListHealthChecksRequest";
const _LHCRi = "ListHealthChecksResponse";
const _LHZ = "ListHostedZones";
const _LHZBN = "ListHostedZonesByName";
const _LHZBNR = "ListHostedZonesByNameRequest";
const _LHZBNRi = "ListHostedZonesByNameResponse";
const _LHZBVPC = "ListHostedZonesByVPC";
const _LHZBVPCR = "ListHostedZonesByVPCRequest";
const _LHZBVPCRi = "ListHostedZonesByVPCResponse";
const _LHZR = "ListHostedZonesRequest";
const _LHZRi = "ListHostedZonesResponse";
const _LMD = "LastModifiedDate";
const _LN = "LocationName";
const _LQLC = "ListQueryLoggingConfigs";
const _LQLCR = "ListQueryLoggingConfigsRequest";
const _LQLCRi = "ListQueryLoggingConfigsResponse";
const _LRDS = "ListReusableDelegationSets";
const _LRDSR = "ListReusableDelegationSetsRequest";
const _LRDSRi = "ListReusableDelegationSetsResponse";
const _LRRS = "ListResourceRecordSets";
const _LRRSR = "ListResourceRecordSetsRequest";
const _LRRSRi = "ListResourceRecordSetsResponse";
const _LS = "LinkedService";
const _LSo = "LocationSummary";
const _LSoc = "LocationSummaries";
const _LTFR = "ListTagsForResource";
const _LTFRR = "ListTagsForResourceRequest";
const _LTFRRi = "ListTagsForResourceResponse";
const _LTFRRis = "ListTagsForResourcesRequest";
const _LTFRRist = "ListTagsForResourcesResponse";
const _LTFRi = "ListTagsForResources";
const _LTP = "ListTrafficPolicies";
const _LTPI = "ListTrafficPolicyInstances";
const _LTPIBHZ = "ListTrafficPolicyInstancesByHostedZone";
const _LTPIBHZR = "ListTrafficPolicyInstancesByHostedZoneRequest";
const _LTPIBHZRi = "ListTrafficPolicyInstancesByHostedZoneResponse";
const _LTPIBP = "ListTrafficPolicyInstancesByPolicy";
const _LTPIBPR = "ListTrafficPolicyInstancesByPolicyRequest";
const _LTPIBPRi = "ListTrafficPolicyInstancesByPolicyResponse";
const _LTPIR = "ListTrafficPolicyInstancesRequest";
const _LTPIRi = "ListTrafficPolicyInstancesResponse";
const _LTPR = "ListTrafficPoliciesRequest";
const _LTPRi = "ListTrafficPoliciesResponse";
const _LTPV = "ListTrafficPolicyVersions";
const _LTPVR = "ListTrafficPolicyVersionsRequest";
const _LTPVRi = "ListTrafficPolicyVersionsResponse";
const _LV = "LatestVersion";
const _LVPCA = "LastVPCAssociation";
const _LVPCAA = "ListVPCAssociationAuthorizations";
const _LVPCAAR = "ListVPCAssociationAuthorizationsRequest";
const _LVPCAARi = "ListVPCAssociationAuthorizationsResponse";
const _LZG = "LocalZoneGroup";
const _Li = "Limit";
const _Lo = "Longitude";
const _Loc = "Location";
const _M = "Message";
const _MI = "MaxItems";
const _ML = "MeasureLatency";
const _MN = "MetricName";
const _MR = "MaxResults";
const _MVA = "MultiValueAnswer";
const _Ma = "Marker";
const _N = "Name";
const _NAE = "NotAuthorizedException";
const _NCC = "NextContinentCode";
const _NCCe = "NextCountryCode";
const _NDNSN = "NextDNSName";
const _NHZI = "NextHostedZoneId";
const _NM = "NextMarker";
const _NRI = "NextRecordIdentifier";
const _NRN = "NextRecordName";
const _NRT = "NextRecordType";
const _NS = "NameServers";
const _NSC = "NextSubdivisionCode";
const _NSCCE = "NoSuchCidrCollectionException";
const _NSCLE = "NoSuchCidrLocationException";
const _NSCWLLG = "NoSuchCloudWatchLogsLogGroup";
const _NSCo = "NoSuchChange";
const _NSDS = "NoSuchDelegationSet";
const _NSGL = "NoSuchGeoLocation";
const _NSHC = "NoSuchHealthCheck";
const _NSHZ = "NoSuchHostedZone";
const _NSKSK = "NoSuchKeySigningKey";
const _NSQLC = "NoSuchQueryLoggingConfig";
const _NSTP = "NoSuchTrafficPolicy";
const _NSTPI = "NoSuchTrafficPolicyInstance";
const _NSa = "NameServer";
const _NT = "NextToken";
const _Na = "Namespace";
const _Nam = "Nameserver";
const _O = "Owner";
const _OA = "OwningAccount";
const _OS = "OwningService";
const _P = "Period";
const _PK = "PublicKey";
const _PRNC = "PriorRequestNotComplete";
const _PZ = "PrivateZone";
const _PZVPCA = "PublicZoneVPCAssociation";
const _Po = "Port";
const _Pr = "Protocol";
const _QLC = "QueryLoggingConfig";
const _QLCAE = "QueryLoggingConfigAlreadyExists";
const _QLCu = "QueryLoggingConfigs";
const _R = "Region";
const _RC = "ResponseCode";
const _RCA = "RoutingControlArn";
const _RD = "RecordData";
const _RDE = "RecordDataEntry";
const _RDSL = "ReusableDelegationSetLimit";
const _RE = "ResetElements";
const _REN = "ResettableElementName";
const _RENL = "ResettableElementNameList";
const _RI = "ResourceId";
const _RIP = "ResolverIP";
const _RIe = "RequestInterval";
const _RIes = "ResourceIds";
const _RN = "RecordName";
const _RP = "ResourcePath";
const _RR = "ResourceRecord";
const _RRS = "ResourceRecordSet";
const _RRSC = "ResourceRecordSetCount";
const _RRSe = "ResourceRecordSets";
const _RRe = "ResourceRecords";
const _RT = "ResourceType";
const _RTK = "RemoveTagKeys";
const _RTS = "ResourceTagSet";
const _RTSL = "ResourceTagSetList";
const _RTSe = "ResourceTagSets";
const _RTe = "RecordType";
const _Re = "Regions";
const _S = "Status";
const _SA = "SubmittedAt";
const _SAM = "SigningAlgorithmMnemonic";
const _SAT = "SigningAlgorithmType";
const _SC = "SubdivisionCode";
const _SCC = "StartContinentCode";
const _SCCt = "StartCountryCode";
const _SI = "SetIdentifier";
const _SM = "StatusMessage";
const _SN = "SubdivisionName";
const _SP = "ServicePrincipal";
const _SR = "StatusReport";
const _SRI = "StartRecordIdentifier";
const _SRN = "StartRecordName";
const _SRT = "StartRecordType";
const _SS = "ServeSignature";
const _SSC = "StartSubdivisionCode";
const _SSe = "SearchString";
const _St = "Statistic";
const _Sta = "State";
const _T = "Type";
const _TDNSA = "TestDNSAnswer";
const _TDNSAR = "TestDNSAnswerRequest";
const _TDNSARe = "TestDNSAnswerResponse";
const _TE = "ThrottlingException";
const _TKL = "TagKeyList";
const _TL = "TagList";
const _TMHC = "TooManyHealthChecks";
const _TMHZ = "TooManyHostedZones";
const _TMKSK = "TooManyKeySigningKeys";
const _TMTP = "TooManyTrafficPolicies";
const _TMTPI = "TooManyTrafficPolicyInstances";
const _TMTPVFCP = "TooManyTrafficPolicyVersionsForCurrentPolicy";
const _TMVPCAA = "TooManyVPCAssociationAuthorizations";
const _TP = "TrafficPolicy";
const _TPAE = "TrafficPolicyAlreadyExists";
const _TPC = "TrafficPolicyCount";
const _TPI = "TrafficPolicyId";
const _TPIAE = "TrafficPolicyInstanceAlreadyExists";
const _TPIC = "TrafficPolicyInstanceCount";
const _TPII = "TrafficPolicyInstanceId";
const _TPIM = "TrafficPolicyIdMarker";
const _TPINM = "TrafficPolicyInstanceNameMarker";
const _TPITM = "TrafficPolicyInstanceTypeMarker";
const _TPIU = "TrafficPolicyInUse";
const _TPIr = "TrafficPolicyInstance";
const _TPIra = "TrafficPolicyInstances";
const _TPS = "TrafficPolicySummaries";
const _TPSr = "TrafficPolicySummary";
const _TPT = "TrafficPolicyType";
const _TPV = "TrafficPolicyVersion";
const _TPVM = "TrafficPolicyVersionMarker";
const _TPr = "TrafficPolicies";
const _TRIL = "TagResourceIdList";
const _TTL = "TTL";
const _Ta = "Tags";
const _Tag = "Tag";
const _Th = "Threshold";
const _UHC = "UpdateHealthCheck";
const _UHCR = "UpdateHealthCheckRequest";
const _UHCRp = "UpdateHealthCheckResponse";
const _UHZC = "UpdateHostedZoneComment";
const _UHZCR = "UpdateHostedZoneCommentRequest";
const _UHZCRp = "UpdateHostedZoneCommentResponse";
const _UTPC = "UpdateTrafficPolicyComment";
const _UTPCR = "UpdateTrafficPolicyCommentRequest";
const _UTPCRp = "UpdateTrafficPolicyCommentResponse";
const _UTPI = "UpdateTrafficPolicyInstance";
const _UTPIR = "UpdateTrafficPolicyInstanceRequest";
const _UTPIRp = "UpdateTrafficPolicyInstanceResponse";
const _V = "Value";
const _VPC = "VPC";
const _VPCAANF = "VPCAssociationAuthorizationNotFound";
const _VPCANF = "VPCAssociationNotFound";
const _VPCI = "VPCId";
const _VPCR = "VPCRegion";
const _VPCs = "VPCs";
const _Ve = "Version";
const _W = "Weight";
const _c = "client";
const _co = "continentcode";
const _cou = "countrycode";
const _d = "dnsname";
const _de = "delegationsetid";
const _e = "error";
const _ed = "edns0clientsubnetip";
const _edn = "edns0clientsubnetmask";
const _h = "hostedzoneid";
const _hE = "httpError";
const _hH = "httpHeader";
const _hQ = "httpQuery";
const _ho = "hostedzonetype";
const _ht = "http";
const _i = "identifier";
const _id = "id";
const _l = "location";
const _m = "message";
const _ma = "maxresults";
const _mar = "marker";
const _max = "maxitems";
const _me = "messages";
const _n = "nexttoken";
const _na = "name";
const _r = "recordname";
const _re = "recordtype";
const _res = "resolverip";
const _s = "subdivisioncode";
const _sm = "smithy.ts.sdk.synthetic.com.amazonaws.route53";
const _st = "startcontinentcode";
const _sta = "startcountrycode";
const _star = "startsubdivisioncode";
const _t = "type";
const _tr = "trafficpolicyid";
const _tra = "trafficpolicyinstancename";
const _traf = "trafficpolicyinstancetype";
const _traff = "trafficpolicyversion";
const _v = "vpcid";
const _ve = "version";
const _vp = "vpcregion";
const _xN = "xmlName";
const n0 = "com.amazonaws.route53";
var AccountLimit = [3, n0, _AL, 0, [_T, _V], [0, 1]];
var ActivateKeySigningKeyRequest = [
    3,
    n0,
    _AKSKR,
    0,
    [_HZI, _N],
    [
        [0, 1],
        [0, 1],
    ],
];
var ActivateKeySigningKeyResponse = [3, n0, _AKSKRc, 0, [_CI], [() => ChangeInfo]];
var AlarmIdentifier = [3, n0, _AI, 0, [_R, _N], [0, 0]];
var AliasTarget = [3, n0, _AT, 0, [_HZI, _DNSN, _ETH], [0, 0, 2]];
var AssociateVPCWithHostedZoneRequest = [
    3,
    n0,
    _AVPCWHZR,
    0,
    [_HZI, _VPC, _C],
    [[0, 1], () => VPC, 0],
];
var AssociateVPCWithHostedZoneResponse = [
    3,
    n0,
    _AVPCWHZRs,
    0,
    [_CI],
    [() => ChangeInfo],
];
var Change = [3, n0, _Ch, 0, [_A, _RRS], [0, [() => ResourceRecordSet, 0]]];
var ChangeBatch = [3, n0, _CB, 0, [_C, _Cha], [0, [() => Changes, 0]]];
var ChangeCidrCollectionRequest = [
    3,
    n0,
    _CCCR,
    0,
    [_I, _CV, _Cha],
    [[0, 1], 1, [() => CidrCollectionChanges, 0]],
];
var ChangeCidrCollectionResponse = [3, n0, _CCCRh, 0, [_I], [0]];
var ChangeInfo = [3, n0, _CI, 0, [_I, _S, _SA, _C], [0, 0, 4, 0]];
var ChangeResourceRecordSetsRequest = [
    3,
    n0,
    _CRRSR,
    0,
    [_HZI, _CB],
    [
        [0, 1],
        [() => ChangeBatch, 0],
    ],
];
var ChangeResourceRecordSetsResponse = [3, n0, _CRRSRh, 0, [_CI], [() => ChangeInfo]];
var ChangeTagsForResourceRequest = [
    3,
    n0,
    _CTFRR,
    0,
    [_RT, _RI, _ATd, _RTK],
    [
        [0, 1],
        [0, 1],
        [() => TagList, 0],
        [() => TagKeyList, 0],
    ],
];
var ChangeTagsForResourceResponse = [3, n0, _CTFRRh, 0, [], []];
var CidrBlockInUseException = [
    -3,
    n0,
    _CBIUE,
    {
        [_e]: _c,
        [_hE]: 400,
    },
    [_M],
    [0],
];
schema.TypeRegistry.for(n0).registerError(CidrBlockInUseException, CidrBlockInUseException$1);
var CidrBlockSummary = [3, n0, _CBS, 0, [_CBi, _LN], [0, 0]];
var CidrCollection = [3, n0, _CC, 0, [_Ar, _I, _N, _Ve], [0, 0, 0, 1]];
var CidrCollectionAlreadyExistsException = [
    -3,
    n0,
    _CCAEE,
    {
        [_e]: _c,
    },
    [_M],
    [0],
];
schema.TypeRegistry.for(n0).registerError(CidrCollectionAlreadyExistsException, CidrCollectionAlreadyExistsException$1);
var CidrCollectionChange = [3, n0, _CCC, 0, [_LN, _A, _CL], [0, 0, [() => CidrList, 0]]];
var CidrCollectionInUseException = [
    -3,
    n0,
    _CCIUE,
    {
        [_e]: _c,
        [_hE]: 400,
    },
    [_M],
    [0],
];
schema.TypeRegistry.for(n0).registerError(CidrCollectionInUseException, CidrCollectionInUseException$1);
var CidrCollectionVersionMismatchException = [
    -3,
    n0,
    _CCVME,
    {
        [_e]: _c,
        [_hE]: 409,
    },
    [_M],
    [0],
];
schema.TypeRegistry.for(n0).registerError(CidrCollectionVersionMismatchException, CidrCollectionVersionMismatchException$1);
var CidrRoutingConfig = [3, n0, _CRC, 0, [_CIo, _LN], [0, 0]];
var CloudWatchAlarmConfiguration = [
    3,
    n0,
    _CWAC,
    0,
    [_EP, _Th, _CO, _P, _MN, _Na, _St, _D],
    [1, 1, 0, 1, 0, 0, 0, [() => DimensionList, 0]],
];
var CollectionSummary = [3, n0, _CS, 0, [_Ar, _I, _N, _Ve], [0, 0, 0, 1]];
var ConcurrentModification = [
    -3,
    n0,
    _CM,
    {
        [_e]: _c,
        [_hE]: 400,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(ConcurrentModification, ConcurrentModification$1);
var ConflictingDomainExists = [
    -3,
    n0,
    _CDE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(ConflictingDomainExists, ConflictingDomainExists$1);
var ConflictingTypes = [
    -3,
    n0,
    _CT,
    {
        [_e]: _c,
        [_hE]: 400,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(ConflictingTypes, ConflictingTypes$1);
var Coordinates = [3, n0, _Co, 0, [_L, _Lo], [0, 0]];
var CreateCidrCollectionRequest = [3, n0, _CCCRr, 0, [_N, _CR], [0, 0]];
var CreateCidrCollectionResponse = [
    3,
    n0,
    _CCCRre,
    0,
    [_Col, _Loc],
    [
        () => CidrCollection,
        [
            0,
            {
                [_hH]: _Loc,
            },
        ],
    ],
];
var CreateHealthCheckRequest = [
    3,
    n0,
    _CHCR,
    0,
    [_CR, _HCC],
    [0, [() => HealthCheckConfig, 0]],
];
var CreateHealthCheckResponse = [
    3,
    n0,
    _CHCRr,
    0,
    [_HC, _Loc],
    [
        [() => HealthCheck, 0],
        [
            0,
            {
                [_hH]: _Loc,
            },
        ],
    ],
];
var CreateHostedZoneRequest = [
    3,
    n0,
    _CHZR,
    0,
    [_N, _VPC, _CR, _HZC, _DSI],
    [0, () => VPC, 0, () => HostedZoneConfig, 0],
];
var CreateHostedZoneResponse = [
    3,
    n0,
    _CHZRr,
    0,
    [_HZ, _CI, _DS, _VPC, _Loc],
    [
        () => HostedZone,
        () => ChangeInfo,
        [() => DelegationSet, 0],
        () => VPC,
        [
            0,
            {
                [_hH]: _Loc,
            },
        ],
    ],
];
var CreateKeySigningKeyRequest = [
    3,
    n0,
    _CKSKR,
    0,
    [_CR, _HZI, _KMSA, _N, _S],
    [0, 0, 0, 0, 0],
];
var CreateKeySigningKeyResponse = [
    3,
    n0,
    _CKSKRr,
    0,
    [_CI, _KSK, _Loc],
    [
        () => ChangeInfo,
        () => KeySigningKey,
        [
            0,
            {
                [_hH]: _Loc,
            },
        ],
    ],
];
var CreateQueryLoggingConfigRequest = [3, n0, _CQLCR, 0, [_HZI, _CWLLGA], [0, 0]];
var CreateQueryLoggingConfigResponse = [
    3,
    n0,
    _CQLCRr,
    0,
    [_QLC, _Loc],
    [
        () => QueryLoggingConfig,
        [
            0,
            {
                [_hH]: _Loc,
            },
        ],
    ],
];
var CreateReusableDelegationSetRequest = [3, n0, _CRDSR, 0, [_CR, _HZI], [0, 0]];
var CreateReusableDelegationSetResponse = [
    3,
    n0,
    _CRDSRr,
    0,
    [_DS, _Loc],
    [
        [() => DelegationSet, 0],
        [
            0,
            {
                [_hH]: _Loc,
            },
        ],
    ],
];
var CreateTrafficPolicyInstanceRequest = [
    3,
    n0,
    _CTPIR,
    0,
    [_HZI, _N, _TTL, _TPI, _TPV],
    [0, 0, 1, 0, 1],
];
var CreateTrafficPolicyInstanceResponse = [
    3,
    n0,
    _CTPIRr,
    0,
    [_TPIr, _Loc],
    [
        () => TrafficPolicyInstance,
        [
            0,
            {
                [_hH]: _Loc,
            },
        ],
    ],
];
var CreateTrafficPolicyRequest = [3, n0, _CTPR, 0, [_N, _Do, _C], [0, 0, 0]];
var CreateTrafficPolicyResponse = [
    3,
    n0,
    _CTPRr,
    0,
    [_TP, _Loc],
    [
        () => TrafficPolicy,
        [
            0,
            {
                [_hH]: _Loc,
            },
        ],
    ],
];
var CreateTrafficPolicyVersionRequest = [3, n0, _CTPVR, 0, [_I, _Do, _C], [[0, 1], 0, 0]];
var CreateTrafficPolicyVersionResponse = [
    3,
    n0,
    _CTPVRr,
    0,
    [_TP, _Loc],
    [
        () => TrafficPolicy,
        [
            0,
            {
                [_hH]: _Loc,
            },
        ],
    ],
];
var CreateVPCAssociationAuthorizationRequest = [
    3,
    n0,
    _CVPCAAR,
    0,
    [_HZI, _VPC],
    [[0, 1], () => VPC],
];
var CreateVPCAssociationAuthorizationResponse = [
    3,
    n0,
    _CVPCAARr,
    0,
    [_HZI, _VPC],
    [0, () => VPC],
];
var DeactivateKeySigningKeyRequest = [
    3,
    n0,
    _DKSKR,
    0,
    [_HZI, _N],
    [
        [0, 1],
        [0, 1],
    ],
];
var DeactivateKeySigningKeyResponse = [3, n0, _DKSKRe, 0, [_CI], [() => ChangeInfo]];
var DelegationSet = [
    3,
    n0,
    _DS,
    0,
    [_I, _CR, _NS],
    [0, 0, [() => DelegationSetNameServers, 0]],
];
var DelegationSetAlreadyCreated = [
    -3,
    n0,
    _DSAC,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(DelegationSetAlreadyCreated, DelegationSetAlreadyCreated$1);
var DelegationSetAlreadyReusable = [
    -3,
    n0,
    _DSAR,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(DelegationSetAlreadyReusable, DelegationSetAlreadyReusable$1);
var DelegationSetInUse = [
    -3,
    n0,
    _DSIU,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(DelegationSetInUse, DelegationSetInUse$1);
var DelegationSetNotAvailable = [
    -3,
    n0,
    _DSNA,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(DelegationSetNotAvailable, DelegationSetNotAvailable$1);
var DelegationSetNotReusable = [
    -3,
    n0,
    _DSNR,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(DelegationSetNotReusable, DelegationSetNotReusable$1);
var DeleteCidrCollectionRequest = [3, n0, _DCCR, 0, [_I], [[0, 1]]];
var DeleteCidrCollectionResponse = [3, n0, _DCCRe, 0, [], []];
var DeleteHealthCheckRequest = [3, n0, _DHCR, 0, [_HCI], [[0, 1]]];
var DeleteHealthCheckResponse = [3, n0, _DHCRe, 0, [], []];
var DeleteHostedZoneRequest = [3, n0, _DHZR, 0, [_I], [[0, 1]]];
var DeleteHostedZoneResponse = [3, n0, _DHZRe, 0, [_CI], [() => ChangeInfo]];
var DeleteKeySigningKeyRequest = [
    3,
    n0,
    _DKSKRel,
    0,
    [_HZI, _N],
    [
        [0, 1],
        [0, 1],
    ],
];
var DeleteKeySigningKeyResponse = [3, n0, _DKSKRele, 0, [_CI], [() => ChangeInfo]];
var DeleteQueryLoggingConfigRequest = [3, n0, _DQLCR, 0, [_I], [[0, 1]]];
var DeleteQueryLoggingConfigResponse = [3, n0, _DQLCRe, 0, [], []];
var DeleteReusableDelegationSetRequest = [3, n0, _DRDSR, 0, [_I], [[0, 1]]];
var DeleteReusableDelegationSetResponse = [3, n0, _DRDSRe, 0, [], []];
var DeleteTrafficPolicyInstanceRequest = [3, n0, _DTPIR, 0, [_I], [[0, 1]]];
var DeleteTrafficPolicyInstanceResponse = [3, n0, _DTPIRe, 0, [], []];
var DeleteTrafficPolicyRequest = [
    3,
    n0,
    _DTPR,
    0,
    [_I, _Ve],
    [
        [0, 1],
        [1, 1],
    ],
];
var DeleteTrafficPolicyResponse = [3, n0, _DTPRe, 0, [], []];
var DeleteVPCAssociationAuthorizationRequest = [
    3,
    n0,
    _DVPCAAR,
    0,
    [_HZI, _VPC],
    [[0, 1], () => VPC],
];
var DeleteVPCAssociationAuthorizationResponse = [3, n0, _DVPCAARe, 0, [], []];
var Dimension = [3, n0, _Di, 0, [_N, _V], [0, 0]];
var DisableHostedZoneDNSSECRequest = [3, n0, _DHZDNSSECR, 0, [_HZI], [[0, 1]]];
var DisableHostedZoneDNSSECResponse = [3, n0, _DHZDNSSECRi, 0, [_CI], [() => ChangeInfo]];
var DisassociateVPCFromHostedZoneRequest = [
    3,
    n0,
    _DVPCFHZR,
    0,
    [_HZI, _VPC, _C],
    [[0, 1], () => VPC, 0],
];
var DisassociateVPCFromHostedZoneResponse = [
    3,
    n0,
    _DVPCFHZRi,
    0,
    [_CI],
    [() => ChangeInfo],
];
var DNSSECNotFound = [
    -3,
    n0,
    _DNSSECNF,
    {
        [_e]: _c,
        [_hE]: 400,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(DNSSECNotFound, DNSSECNotFound$1);
var DNSSECStatus = [3, n0, _DNSSECS, 0, [_SS, _SM], [0, 0]];
var EnableHostedZoneDNSSECRequest = [3, n0, _EHZDNSSECR, 0, [_HZI], [[0, 1]]];
var EnableHostedZoneDNSSECResponse = [3, n0, _EHZDNSSECRn, 0, [_CI], [() => ChangeInfo]];
var GeoLocation = [3, n0, _GL, 0, [_CCo, _CCou, _SC], [0, 0, 0]];
var GeoLocationDetails = [
    3,
    n0,
    _GLD,
    0,
    [_CCo, _CN, _CCou, _CNo, _SC, _SN],
    [0, 0, 0, 0, 0, 0],
];
var GeoProximityLocation = [
    3,
    n0,
    _GPL,
    0,
    [_AWSR, _LZG, _Co, _B],
    [0, 0, () => Coordinates, 1],
];
var GetAccountLimitRequest = [3, n0, _GALR, 0, [_T], [[0, 1]]];
var GetAccountLimitResponse = [3, n0, _GALRe, 0, [_Li, _Cou], [() => AccountLimit, 1]];
var GetChangeRequest = [3, n0, _GCR, 0, [_I], [[0, 1]]];
var GetChangeResponse = [3, n0, _GCRe, 0, [_CI], [() => ChangeInfo]];
var GetCheckerIpRangesRequest = [3, n0, _GCIRR, 0, [], []];
var GetCheckerIpRangesResponse = [3, n0, _GCIRRe, 0, [_CIR], [64 | 0]];
var GetDNSSECRequest = [3, n0, _GDNSSECR, 0, [_HZI], [[0, 1]]];
var GetDNSSECResponse = [
    3,
    n0,
    _GDNSSECRe,
    0,
    [_S, _KSKe],
    [() => DNSSECStatus, () => KeySigningKeys],
];
var GetGeoLocationRequest = [
    3,
    n0,
    _GGLR,
    0,
    [_CCo, _CCou, _SC],
    [
        [
            0,
            {
                [_hQ]: _co,
            },
        ],
        [
            0,
            {
                [_hQ]: _cou,
            },
        ],
        [
            0,
            {
                [_hQ]: _s,
            },
        ],
    ],
];
var GetGeoLocationResponse = [3, n0, _GGLRe, 0, [_GLD], [() => GeoLocationDetails]];
var GetHealthCheckCountRequest = [3, n0, _GHCCR, 0, [], []];
var GetHealthCheckCountResponse = [3, n0, _GHCCRe, 0, [_HCCe], [1]];
var GetHealthCheckLastFailureReasonRequest = [3, n0, _GHCLFRR, 0, [_HCI], [[0, 1]]];
var GetHealthCheckLastFailureReasonResponse = [
    3,
    n0,
    _GHCLFRRe,
    0,
    [_HCO],
    [[() => HealthCheckObservations, 0]],
];
var GetHealthCheckRequest = [3, n0, _GHCR, 0, [_HCI], [[0, 1]]];
var GetHealthCheckResponse = [3, n0, _GHCRe, 0, [_HC], [[() => HealthCheck, 0]]];
var GetHealthCheckStatusRequest = [3, n0, _GHCSR, 0, [_HCI], [[0, 1]]];
var GetHealthCheckStatusResponse = [
    3,
    n0,
    _GHCSRe,
    0,
    [_HCO],
    [[() => HealthCheckObservations, 0]],
];
var GetHostedZoneCountRequest = [3, n0, _GHZCR, 0, [], []];
var GetHostedZoneCountResponse = [3, n0, _GHZCRe, 0, [_HZCo], [1]];
var GetHostedZoneLimitRequest = [
    3,
    n0,
    _GHZLR,
    0,
    [_T, _HZI],
    [
        [0, 1],
        [0, 1],
    ],
];
var GetHostedZoneLimitResponse = [
    3,
    n0,
    _GHZLRe,
    0,
    [_Li, _Cou],
    [() => HostedZoneLimit, 1],
];
var GetHostedZoneRequest = [3, n0, _GHZR, 0, [_I], [[0, 1]]];
var GetHostedZoneResponse = [
    3,
    n0,
    _GHZRe,
    0,
    [_HZ, _DS, _VPCs],
    [() => HostedZone, [() => DelegationSet, 0], [() => VPCs, 0]],
];
var GetQueryLoggingConfigRequest = [3, n0, _GQLCR, 0, [_I], [[0, 1]]];
var GetQueryLoggingConfigResponse = [
    3,
    n0,
    _GQLCRe,
    0,
    [_QLC],
    [() => QueryLoggingConfig],
];
var GetReusableDelegationSetLimitRequest = [
    3,
    n0,
    _GRDSLR,
    0,
    [_T, _DSI],
    [
        [0, 1],
        [0, 1],
    ],
];
var GetReusableDelegationSetLimitResponse = [
    3,
    n0,
    _GRDSLRe,
    0,
    [_Li, _Cou],
    [() => ReusableDelegationSetLimit, 1],
];
var GetReusableDelegationSetRequest = [3, n0, _GRDSR, 0, [_I], [[0, 1]]];
var GetReusableDelegationSetResponse = [
    3,
    n0,
    _GRDSRe,
    0,
    [_DS],
    [[() => DelegationSet, 0]],
];
var GetTrafficPolicyInstanceCountRequest = [3, n0, _GTPICR, 0, [], []];
var GetTrafficPolicyInstanceCountResponse = [3, n0, _GTPICRe, 0, [_TPIC], [1]];
var GetTrafficPolicyInstanceRequest = [3, n0, _GTPIR, 0, [_I], [[0, 1]]];
var GetTrafficPolicyInstanceResponse = [
    3,
    n0,
    _GTPIRe,
    0,
    [_TPIr],
    [() => TrafficPolicyInstance],
];
var GetTrafficPolicyRequest = [
    3,
    n0,
    _GTPR,
    0,
    [_I, _Ve],
    [
        [0, 1],
        [1, 1],
    ],
];
var GetTrafficPolicyResponse = [3, n0, _GTPRe, 0, [_TP], [() => TrafficPolicy]];
var HealthCheck = [
    3,
    n0,
    _HC,
    0,
    [_I, _CR, _LS, _HCC, _HCV, _CWAC],
    [0, 0, () => LinkedService, [() => HealthCheckConfig, 0], 1, [() => CloudWatchAlarmConfiguration, 0]],
];
var HealthCheckAlreadyExists = [
    -3,
    n0,
    _HCAE,
    {
        [_e]: _c,
        [_hE]: 409,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(HealthCheckAlreadyExists, HealthCheckAlreadyExists$1);
var HealthCheckConfig = [
    3,
    n0,
    _HCC,
    0,
    [_IPA, _Po, _T, _RP, _FQDN, _SSe, _RIe, _FT, _ML, _In, _Dis, _HT, _CHC, _ESNI, _Re, _AI, _IDHS, _RCA],
    [
        0,
        1,
        0,
        0,
        0,
        0,
        1,
        1,
        2,
        2,
        2,
        1,
        [() => ChildHealthCheckList, 0],
        2,
        [() => HealthCheckRegionList, 0],
        () => AlarmIdentifier,
        0,
        0,
    ],
];
var HealthCheckInUse = [
    -3,
    n0,
    _HCIU,
    {
        [_e]: _c,
        [_hE]: 400,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(HealthCheckInUse, HealthCheckInUse$1);
var HealthCheckObservation = [
    3,
    n0,
    _HCOe,
    0,
    [_R, _IPA, _SR],
    [0, 0, () => StatusReport],
];
var HealthCheckVersionMismatch = [
    -3,
    n0,
    _HCVM,
    {
        [_e]: _c,
        [_hE]: 409,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(HealthCheckVersionMismatch, HealthCheckVersionMismatch$1);
var HostedZone = [
    3,
    n0,
    _HZ,
    0,
    [_I, _N, _CR, _Con, _RRSC, _LS],
    [0, 0, 0, () => HostedZoneConfig, 1, () => LinkedService],
];
var HostedZoneAlreadyExists = [
    -3,
    n0,
    _HZAE,
    {
        [_e]: _c,
        [_hE]: 409,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(HostedZoneAlreadyExists, HostedZoneAlreadyExists$1);
var HostedZoneConfig = [3, n0, _HZC, 0, [_C, _PZ], [0, 2]];
var HostedZoneLimit = [3, n0, _HZL, 0, [_T, _V], [0, 1]];
var HostedZoneNotEmpty = [
    -3,
    n0,
    _HZNE,
    {
        [_e]: _c,
        [_hE]: 400,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(HostedZoneNotEmpty, HostedZoneNotEmpty$1);
var HostedZoneNotFound = [
    -3,
    n0,
    _HZNF,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(HostedZoneNotFound, HostedZoneNotFound$1);
var HostedZoneNotPrivate = [
    -3,
    n0,
    _HZNP,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(HostedZoneNotPrivate, HostedZoneNotPrivate$1);
var HostedZoneOwner = [3, n0, _HZO, 0, [_OA, _OS], [0, 0]];
var HostedZonePartiallyDelegated = [
    -3,
    n0,
    _HZPD,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(HostedZonePartiallyDelegated, HostedZonePartiallyDelegated$1);
var HostedZoneSummary = [3, n0, _HZS, 0, [_HZI, _N, _O], [0, 0, () => HostedZoneOwner]];
var IncompatibleVersion = [
    -3,
    n0,
    _IV,
    {
        [_e]: _c,
        [_hE]: 400,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(IncompatibleVersion, IncompatibleVersion$1);
var InsufficientCloudWatchLogsResourcePolicy = [
    -3,
    n0,
    _ICWLRP,
    {
        [_e]: _c,
        [_hE]: 400,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(InsufficientCloudWatchLogsResourcePolicy, InsufficientCloudWatchLogsResourcePolicy$1);
var InvalidArgument = [
    -3,
    n0,
    _IA,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(InvalidArgument, InvalidArgument$1);
var InvalidChangeBatch = [
    -3,
    n0,
    _ICB,
    {
        [_e]: _c,
    },
    [_me, _m],
    [[() => ErrorMessages, 0], 0],
];
schema.TypeRegistry.for(n0).registerError(InvalidChangeBatch, InvalidChangeBatch$1);
var InvalidDomainName = [
    -3,
    n0,
    _IDN,
    {
        [_e]: _c,
        [_hE]: 400,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(InvalidDomainName, InvalidDomainName$1);
var InvalidInput = [
    -3,
    n0,
    _II,
    {
        [_e]: _c,
        [_hE]: 400,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(InvalidInput, InvalidInput$1);
var InvalidKeySigningKeyName = [
    -3,
    n0,
    _IKSKN,
    {
        [_e]: _c,
        [_hE]: 400,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(InvalidKeySigningKeyName, InvalidKeySigningKeyName$1);
var InvalidKeySigningKeyStatus = [
    -3,
    n0,
    _IKSKS,
    {
        [_e]: _c,
        [_hE]: 400,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(InvalidKeySigningKeyStatus, InvalidKeySigningKeyStatus$1);
var InvalidKMSArn = [
    -3,
    n0,
    _IKMSA,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(InvalidKMSArn, InvalidKMSArn$1);
var InvalidPaginationToken = [
    -3,
    n0,
    _IPT,
    {
        [_e]: _c,
        [_hE]: 400,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(InvalidPaginationToken, InvalidPaginationToken$1);
var InvalidSigningStatus = [
    -3,
    n0,
    _ISS,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(InvalidSigningStatus, InvalidSigningStatus$1);
var InvalidTrafficPolicyDocument = [
    -3,
    n0,
    _ITPD,
    {
        [_e]: _c,
        [_hE]: 400,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(InvalidTrafficPolicyDocument, InvalidTrafficPolicyDocument$1);
var InvalidVPCId = [
    -3,
    n0,
    _IVPCI,
    {
        [_e]: _c,
        [_hE]: 400,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(InvalidVPCId, InvalidVPCId$1);
var KeySigningKey = [
    3,
    n0,
    _KSK,
    0,
    [_N, _KA, _F, _SAM, _SAT, _DAM, _DAT, _KT, _DV, _PK, _DSR, _DNSKEYR, _S, _SM, _CD, _LMD],
    [0, 0, 1, 0, 1, 0, 1, 1, 0, 0, 0, 0, 0, 0, 4, 4],
];
var KeySigningKeyAlreadyExists = [
    -3,
    n0,
    _KSKAE,
    {
        [_e]: _c,
        [_hE]: 409,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(KeySigningKeyAlreadyExists, KeySigningKeyAlreadyExists$1);
var KeySigningKeyInParentDSRecord = [
    -3,
    n0,
    _KSKIPDSR,
    {
        [_e]: _c,
        [_hE]: 400,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(KeySigningKeyInParentDSRecord, KeySigningKeyInParentDSRecord$1);
var KeySigningKeyInUse = [
    -3,
    n0,
    _KSKIU,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(KeySigningKeyInUse, KeySigningKeyInUse$1);
var KeySigningKeyWithActiveStatusNotFound = [
    -3,
    n0,
    _KSKWASNF,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(KeySigningKeyWithActiveStatusNotFound, KeySigningKeyWithActiveStatusNotFound$1);
var LastVPCAssociation = [
    -3,
    n0,
    _LVPCA,
    {
        [_e]: _c,
        [_hE]: 400,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(LastVPCAssociation, LastVPCAssociation$1);
var LimitsExceeded = [
    -3,
    n0,
    _LE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(LimitsExceeded, LimitsExceeded$1);
var LinkedService = [3, n0, _LS, 0, [_SP, _De], [0, 0]];
var ListCidrBlocksRequest = [
    3,
    n0,
    _LCBR,
    0,
    [_CIo, _LN, _NT, _MR],
    [
        [0, 1],
        [
            0,
            {
                [_hQ]: _l,
            },
        ],
        [
            0,
            {
                [_hQ]: _n,
            },
        ],
        [
            1,
            {
                [_hQ]: _ma,
            },
        ],
    ],
];
var ListCidrBlocksResponse = [
    3,
    n0,
    _LCBRi,
    0,
    [_NT, _CBid],
    [0, () => CidrBlockSummaries],
];
var ListCidrCollectionsRequest = [
    3,
    n0,
    _LCCR,
    0,
    [_NT, _MR],
    [
        [
            0,
            {
                [_hQ]: _n,
            },
        ],
        [
            1,
            {
                [_hQ]: _ma,
            },
        ],
    ],
];
var ListCidrCollectionsResponse = [
    3,
    n0,
    _LCCRi,
    0,
    [_NT, _CCi],
    [0, () => CollectionSummaries],
];
var ListCidrLocationsRequest = [
    3,
    n0,
    _LCLR,
    0,
    [_CIo, _NT, _MR],
    [
        [0, 1],
        [
            0,
            {
                [_hQ]: _n,
            },
        ],
        [
            1,
            {
                [_hQ]: _ma,
            },
        ],
    ],
];
var ListCidrLocationsResponse = [
    3,
    n0,
    _LCLRi,
    0,
    [_NT, _CLi],
    [0, () => LocationSummaries],
];
var ListGeoLocationsRequest = [
    3,
    n0,
    _LGLR,
    0,
    [_SCC, _SCCt, _SSC, _MI],
    [
        [
            0,
            {
                [_hQ]: _st,
            },
        ],
        [
            0,
            {
                [_hQ]: _sta,
            },
        ],
        [
            0,
            {
                [_hQ]: _star,
            },
        ],
        [
            1,
            {
                [_hQ]: _max,
            },
        ],
    ],
];
var ListGeoLocationsResponse = [
    3,
    n0,
    _LGLRi,
    0,
    [_GLDL, _IT, _NCC, _NCCe, _NSC, _MI],
    [[() => GeoLocationDetailsList, 0], 2, 0, 0, 0, 1],
];
var ListHealthChecksRequest = [
    3,
    n0,
    _LHCR,
    0,
    [_Ma, _MI],
    [
        [
            0,
            {
                [_hQ]: _mar,
            },
        ],
        [
            1,
            {
                [_hQ]: _max,
            },
        ],
    ],
];
var ListHealthChecksResponse = [
    3,
    n0,
    _LHCRi,
    0,
    [_HCe, _Ma, _IT, _NM, _MI],
    [[() => HealthChecks, 0], 0, 2, 0, 1],
];
var ListHostedZonesByNameRequest = [
    3,
    n0,
    _LHZBNR,
    0,
    [_DNSN, _HZI, _MI],
    [
        [
            0,
            {
                [_hQ]: _d,
            },
        ],
        [
            0,
            {
                [_hQ]: _h,
            },
        ],
        [
            1,
            {
                [_hQ]: _max,
            },
        ],
    ],
];
var ListHostedZonesByNameResponse = [
    3,
    n0,
    _LHZBNRi,
    0,
    [_HZo, _DNSN, _HZI, _IT, _NDNSN, _NHZI, _MI],
    [[() => HostedZones, 0], 0, 0, 2, 0, 0, 1],
];
var ListHostedZonesByVPCRequest = [
    3,
    n0,
    _LHZBVPCR,
    0,
    [_VPCI, _VPCR, _MI, _NT],
    [
        [
            0,
            {
                [_hQ]: _v,
            },
        ],
        [
            0,
            {
                [_hQ]: _vp,
            },
        ],
        [
            1,
            {
                [_hQ]: _max,
            },
        ],
        [
            0,
            {
                [_hQ]: _n,
            },
        ],
    ],
];
var ListHostedZonesByVPCResponse = [
    3,
    n0,
    _LHZBVPCRi,
    0,
    [_HZSo, _MI, _NT],
    [[() => HostedZoneSummaries, 0], 1, 0],
];
var ListHostedZonesRequest = [
    3,
    n0,
    _LHZR,
    0,
    [_Ma, _MI, _DSI, _HZT],
    [
        [
            0,
            {
                [_hQ]: _mar,
            },
        ],
        [
            1,
            {
                [_hQ]: _max,
            },
        ],
        [
            0,
            {
                [_hQ]: _de,
            },
        ],
        [
            0,
            {
                [_hQ]: _ho,
            },
        ],
    ],
];
var ListHostedZonesResponse = [
    3,
    n0,
    _LHZRi,
    0,
    [_HZo, _Ma, _IT, _NM, _MI],
    [[() => HostedZones, 0], 0, 2, 0, 1],
];
var ListQueryLoggingConfigsRequest = [
    3,
    n0,
    _LQLCR,
    0,
    [_HZI, _NT, _MR],
    [
        [
            0,
            {
                [_hQ]: _h,
            },
        ],
        [
            0,
            {
                [_hQ]: _n,
            },
        ],
        [
            1,
            {
                [_hQ]: _ma,
            },
        ],
    ],
];
var ListQueryLoggingConfigsResponse = [
    3,
    n0,
    _LQLCRi,
    0,
    [_QLCu, _NT],
    [[() => QueryLoggingConfigs, 0], 0],
];
var ListResourceRecordSetsRequest = [
    3,
    n0,
    _LRRSR,
    0,
    [_HZI, _SRN, _SRT, _SRI, _MI],
    [
        [0, 1],
        [
            0,
            {
                [_hQ]: _na,
            },
        ],
        [
            0,
            {
                [_hQ]: _t,
            },
        ],
        [
            0,
            {
                [_hQ]: _i,
            },
        ],
        [
            1,
            {
                [_hQ]: _max,
            },
        ],
    ],
];
var ListResourceRecordSetsResponse = [
    3,
    n0,
    _LRRSRi,
    0,
    [_RRSe, _IT, _NRN, _NRT, _NRI, _MI],
    [[() => ResourceRecordSets, 0], 2, 0, 0, 0, 1],
];
var ListReusableDelegationSetsRequest = [
    3,
    n0,
    _LRDSR,
    0,
    [_Ma, _MI],
    [
        [
            0,
            {
                [_hQ]: _mar,
            },
        ],
        [
            1,
            {
                [_hQ]: _max,
            },
        ],
    ],
];
var ListReusableDelegationSetsResponse = [
    3,
    n0,
    _LRDSRi,
    0,
    [_DSe, _Ma, _IT, _NM, _MI],
    [[() => DelegationSets, 0], 0, 2, 0, 1],
];
var ListTagsForResourceRequest = [
    3,
    n0,
    _LTFRR,
    0,
    [_RT, _RI],
    [
        [0, 1],
        [0, 1],
    ],
];
var ListTagsForResourceResponse = [
    3,
    n0,
    _LTFRRi,
    0,
    [_RTS],
    [[() => ResourceTagSet, 0]],
];
var ListTagsForResourcesRequest = [
    3,
    n0,
    _LTFRRis,
    0,
    [_RT, _RIes],
    [
        [0, 1],
        [() => TagResourceIdList, 0],
    ],
];
var ListTagsForResourcesResponse = [
    3,
    n0,
    _LTFRRist,
    0,
    [_RTSe],
    [[() => ResourceTagSetList, 0]],
];
var ListTrafficPoliciesRequest = [
    3,
    n0,
    _LTPR,
    0,
    [_TPIM, _MI],
    [
        [
            0,
            {
                [_hQ]: _tr,
            },
        ],
        [
            1,
            {
                [_hQ]: _max,
            },
        ],
    ],
];
var ListTrafficPoliciesResponse = [
    3,
    n0,
    _LTPRi,
    0,
    [_TPS, _IT, _TPIM, _MI],
    [[() => TrafficPolicySummaries, 0], 2, 0, 1],
];
var ListTrafficPolicyInstancesByHostedZoneRequest = [
    3,
    n0,
    _LTPIBHZR,
    0,
    [_HZI, _TPINM, _TPITM, _MI],
    [
        [
            0,
            {
                [_hQ]: _id,
            },
        ],
        [
            0,
            {
                [_hQ]: _tra,
            },
        ],
        [
            0,
            {
                [_hQ]: _traf,
            },
        ],
        [
            1,
            {
                [_hQ]: _max,
            },
        ],
    ],
];
var ListTrafficPolicyInstancesByHostedZoneResponse = [
    3,
    n0,
    _LTPIBHZRi,
    0,
    [_TPIra, _TPINM, _TPITM, _IT, _MI],
    [[() => TrafficPolicyInstances, 0], 0, 0, 2, 1],
];
var ListTrafficPolicyInstancesByPolicyRequest = [
    3,
    n0,
    _LTPIBPR,
    0,
    [_TPI, _TPV, _HZIM, _TPINM, _TPITM, _MI],
    [
        [
            0,
            {
                [_hQ]: _id,
            },
        ],
        [
            1,
            {
                [_hQ]: _ve,
            },
        ],
        [
            0,
            {
                [_hQ]: _h,
            },
        ],
        [
            0,
            {
                [_hQ]: _tra,
            },
        ],
        [
            0,
            {
                [_hQ]: _traf,
            },
        ],
        [
            1,
            {
                [_hQ]: _max,
            },
        ],
    ],
];
var ListTrafficPolicyInstancesByPolicyResponse = [
    3,
    n0,
    _LTPIBPRi,
    0,
    [_TPIra, _HZIM, _TPINM, _TPITM, _IT, _MI],
    [[() => TrafficPolicyInstances, 0], 0, 0, 0, 2, 1],
];
var ListTrafficPolicyInstancesRequest = [
    3,
    n0,
    _LTPIR,
    0,
    [_HZIM, _TPINM, _TPITM, _MI],
    [
        [
            0,
            {
                [_hQ]: _h,
            },
        ],
        [
            0,
            {
                [_hQ]: _tra,
            },
        ],
        [
            0,
            {
                [_hQ]: _traf,
            },
        ],
        [
            1,
            {
                [_hQ]: _max,
            },
        ],
    ],
];
var ListTrafficPolicyInstancesResponse = [
    3,
    n0,
    _LTPIRi,
    0,
    [_TPIra, _HZIM, _TPINM, _TPITM, _IT, _MI],
    [[() => TrafficPolicyInstances, 0], 0, 0, 0, 2, 1],
];
var ListTrafficPolicyVersionsRequest = [
    3,
    n0,
    _LTPVR,
    0,
    [_I, _TPVM, _MI],
    [
        [0, 1],
        [
            0,
            {
                [_hQ]: _traff,
            },
        ],
        [
            1,
            {
                [_hQ]: _max,
            },
        ],
    ],
];
var ListTrafficPolicyVersionsResponse = [
    3,
    n0,
    _LTPVRi,
    0,
    [_TPr, _IT, _TPVM, _MI],
    [[() => TrafficPolicies, 0], 2, 0, 1],
];
var ListVPCAssociationAuthorizationsRequest = [
    3,
    n0,
    _LVPCAAR,
    0,
    [_HZI, _NT, _MR],
    [
        [0, 1],
        [
            0,
            {
                [_hQ]: _n,
            },
        ],
        [
            1,
            {
                [_hQ]: _ma,
            },
        ],
    ],
];
var ListVPCAssociationAuthorizationsResponse = [
    3,
    n0,
    _LVPCAARi,
    0,
    [_HZI, _NT, _VPCs],
    [0, 0, [() => VPCs, 0]],
];
var LocationSummary = [3, n0, _LSo, 0, [_LN], [0]];
var NoSuchChange = [
    -3,
    n0,
    _NSCo,
    {
        [_e]: _c,
        [_hE]: 404,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(NoSuchChange, NoSuchChange$1);
var NoSuchCidrCollectionException = [
    -3,
    n0,
    _NSCCE,
    {
        [_e]: _c,
        [_hE]: 404,
    },
    [_M],
    [0],
];
schema.TypeRegistry.for(n0).registerError(NoSuchCidrCollectionException, NoSuchCidrCollectionException$1);
var NoSuchCidrLocationException = [
    -3,
    n0,
    _NSCLE,
    {
        [_e]: _c,
        [_hE]: 404,
    },
    [_M],
    [0],
];
schema.TypeRegistry.for(n0).registerError(NoSuchCidrLocationException, NoSuchCidrLocationException$1);
var NoSuchCloudWatchLogsLogGroup = [
    -3,
    n0,
    _NSCWLLG,
    {
        [_e]: _c,
        [_hE]: 404,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(NoSuchCloudWatchLogsLogGroup, NoSuchCloudWatchLogsLogGroup$1);
var NoSuchDelegationSet = [
    -3,
    n0,
    _NSDS,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(NoSuchDelegationSet, NoSuchDelegationSet$1);
var NoSuchGeoLocation = [
    -3,
    n0,
    _NSGL,
    {
        [_e]: _c,
        [_hE]: 404,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(NoSuchGeoLocation, NoSuchGeoLocation$1);
var NoSuchHealthCheck = [
    -3,
    n0,
    _NSHC,
    {
        [_e]: _c,
        [_hE]: 404,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(NoSuchHealthCheck, NoSuchHealthCheck$1);
var NoSuchHostedZone = [
    -3,
    n0,
    _NSHZ,
    {
        [_e]: _c,
        [_hE]: 404,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(NoSuchHostedZone, NoSuchHostedZone$1);
var NoSuchKeySigningKey = [
    -3,
    n0,
    _NSKSK,
    {
        [_e]: _c,
        [_hE]: 404,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(NoSuchKeySigningKey, NoSuchKeySigningKey$1);
var NoSuchQueryLoggingConfig = [
    -3,
    n0,
    _NSQLC,
    {
        [_e]: _c,
        [_hE]: 404,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(NoSuchQueryLoggingConfig, NoSuchQueryLoggingConfig$1);
var NoSuchTrafficPolicy = [
    -3,
    n0,
    _NSTP,
    {
        [_e]: _c,
        [_hE]: 404,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(NoSuchTrafficPolicy, NoSuchTrafficPolicy$1);
var NoSuchTrafficPolicyInstance = [
    -3,
    n0,
    _NSTPI,
    {
        [_e]: _c,
        [_hE]: 404,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(NoSuchTrafficPolicyInstance, NoSuchTrafficPolicyInstance$1);
var NotAuthorizedException = [
    -3,
    n0,
    _NAE,
    {
        [_e]: _c,
        [_hE]: 401,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(NotAuthorizedException, NotAuthorizedException$1);
var PriorRequestNotComplete = [
    -3,
    n0,
    _PRNC,
    {
        [_e]: _c,
        [_hE]: 400,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(PriorRequestNotComplete, PriorRequestNotComplete$1);
var PublicZoneVPCAssociation = [
    -3,
    n0,
    _PZVPCA,
    {
        [_e]: _c,
        [_hE]: 400,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(PublicZoneVPCAssociation, PublicZoneVPCAssociation$1);
var QueryLoggingConfig = [3, n0, _QLC, 0, [_I, _HZI, _CWLLGA], [0, 0, 0]];
var QueryLoggingConfigAlreadyExists = [
    -3,
    n0,
    _QLCAE,
    {
        [_e]: _c,
        [_hE]: 409,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(QueryLoggingConfigAlreadyExists, QueryLoggingConfigAlreadyExists$1);
var ResourceRecord = [3, n0, _RR, 0, [_V], [0]];
var ResourceRecordSet = [
    3,
    n0,
    _RRS,
    0,
    [_N, _T, _SI, _W, _R, _GL, _Fa, _MVA, _TTL, _RRe, _AT, _HCI, _TPII, _CRC, _GPL],
    [
        0,
        0,
        0,
        1,
        0,
        () => GeoLocation,
        0,
        2,
        1,
        [() => ResourceRecords, 0],
        () => AliasTarget,
        0,
        0,
        () => CidrRoutingConfig,
        () => GeoProximityLocation,
    ],
];
var ResourceTagSet = [3, n0, _RTS, 0, [_RT, _RI, _Ta], [0, 0, [() => TagList, 0]]];
var ReusableDelegationSetLimit = [3, n0, _RDSL, 0, [_T, _V], [0, 1]];
var StatusReport = [3, n0, _SR, 0, [_S, _CTh], [0, 4]];
var Tag = [3, n0, _Tag, 0, [_K, _V], [0, 0]];
var TestDNSAnswerRequest = [
    3,
    n0,
    _TDNSAR,
    0,
    [_HZI, _RN, _RTe, _RIP, _EDNSCSIP, _EDNSCSM],
    [
        [
            0,
            {
                [_hQ]: _h,
            },
        ],
        [
            0,
            {
                [_hQ]: _r,
            },
        ],
        [
            0,
            {
                [_hQ]: _re,
            },
        ],
        [
            0,
            {
                [_hQ]: _res,
            },
        ],
        [
            0,
            {
                [_hQ]: _ed,
            },
        ],
        [
            0,
            {
                [_hQ]: _edn,
            },
        ],
    ],
];
var TestDNSAnswerResponse = [
    3,
    n0,
    _TDNSARe,
    0,
    [_Nam, _RN, _RTe, _RD, _RC, _Pr],
    [0, 0, 0, [() => RecordData, 0], 0, 0],
];
var ThrottlingException = [
    -3,
    n0,
    _TE,
    {
        [_e]: _c,
        [_hE]: 400,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(ThrottlingException, ThrottlingException$1);
var TooManyHealthChecks = [
    -3,
    n0,
    _TMHC,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(TooManyHealthChecks, TooManyHealthChecks$1);
var TooManyHostedZones = [
    -3,
    n0,
    _TMHZ,
    {
        [_e]: _c,
        [_hE]: 400,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(TooManyHostedZones, TooManyHostedZones$1);
var TooManyKeySigningKeys = [
    -3,
    n0,
    _TMKSK,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(TooManyKeySigningKeys, TooManyKeySigningKeys$1);
var TooManyTrafficPolicies = [
    -3,
    n0,
    _TMTP,
    {
        [_e]: _c,
        [_hE]: 400,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(TooManyTrafficPolicies, TooManyTrafficPolicies$1);
var TooManyTrafficPolicyInstances = [
    -3,
    n0,
    _TMTPI,
    {
        [_e]: _c,
        [_hE]: 400,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(TooManyTrafficPolicyInstances, TooManyTrafficPolicyInstances$1);
var TooManyTrafficPolicyVersionsForCurrentPolicy = [
    -3,
    n0,
    _TMTPVFCP,
    {
        [_e]: _c,
        [_hE]: 400,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(TooManyTrafficPolicyVersionsForCurrentPolicy, TooManyTrafficPolicyVersionsForCurrentPolicy$1);
var TooManyVPCAssociationAuthorizations = [
    -3,
    n0,
    _TMVPCAA,
    {
        [_e]: _c,
        [_hE]: 400,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(TooManyVPCAssociationAuthorizations, TooManyVPCAssociationAuthorizations$1);
var TrafficPolicy = [3, n0, _TP, 0, [_I, _Ve, _N, _T, _Do, _C], [0, 1, 0, 0, 0, 0]];
var TrafficPolicyAlreadyExists = [
    -3,
    n0,
    _TPAE,
    {
        [_e]: _c,
        [_hE]: 409,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(TrafficPolicyAlreadyExists, TrafficPolicyAlreadyExists$1);
var TrafficPolicyInstance = [
    3,
    n0,
    _TPIr,
    0,
    [_I, _HZI, _N, _TTL, _Sta, _M, _TPI, _TPV, _TPT],
    [0, 0, 0, 1, 0, 0, 0, 1, 0],
];
var TrafficPolicyInstanceAlreadyExists = [
    -3,
    n0,
    _TPIAE,
    {
        [_e]: _c,
        [_hE]: 409,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(TrafficPolicyInstanceAlreadyExists, TrafficPolicyInstanceAlreadyExists$1);
var TrafficPolicyInUse = [
    -3,
    n0,
    _TPIU,
    {
        [_e]: _c,
        [_hE]: 400,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(TrafficPolicyInUse, TrafficPolicyInUse$1);
var TrafficPolicySummary = [3, n0, _TPSr, 0, [_I, _N, _T, _LV, _TPC], [0, 0, 0, 1, 1]];
var UpdateHealthCheckRequest = [
    3,
    n0,
    _UHCR,
    0,
    [_HCI, _HCV, _IPA, _Po, _RP, _FQDN, _SSe, _FT, _In, _Dis, _HT, _CHC, _ESNI, _Re, _AI, _IDHS, _RE],
    [
        [0, 1],
        1,
        0,
        1,
        0,
        0,
        0,
        1,
        2,
        2,
        1,
        [() => ChildHealthCheckList, 0],
        2,
        [() => HealthCheckRegionList, 0],
        () => AlarmIdentifier,
        0,
        [() => ResettableElementNameList, 0],
    ],
];
var UpdateHealthCheckResponse = [3, n0, _UHCRp, 0, [_HC], [[() => HealthCheck, 0]]];
var UpdateHostedZoneCommentRequest = [3, n0, _UHZCR, 0, [_I, _C], [[0, 1], 0]];
var UpdateHostedZoneCommentResponse = [3, n0, _UHZCRp, 0, [_HZ], [() => HostedZone]];
var UpdateTrafficPolicyCommentRequest = [
    3,
    n0,
    _UTPCR,
    0,
    [_I, _Ve, _C],
    [[0, 1], [1, 1], 0],
];
var UpdateTrafficPolicyCommentResponse = [
    3,
    n0,
    _UTPCRp,
    0,
    [_TP],
    [() => TrafficPolicy],
];
var UpdateTrafficPolicyInstanceRequest = [
    3,
    n0,
    _UTPIR,
    0,
    [_I, _TTL, _TPI, _TPV],
    [[0, 1], 1, 0, 1],
];
var UpdateTrafficPolicyInstanceResponse = [
    3,
    n0,
    _UTPIRp,
    0,
    [_TPIr],
    [() => TrafficPolicyInstance],
];
var VPC = [3, n0, _VPC, 0, [_VPCR, _VPCI], [0, 0]];
var VPCAssociationAuthorizationNotFound = [
    -3,
    n0,
    _VPCAANF,
    {
        [_e]: _c,
        [_hE]: 404,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(VPCAssociationAuthorizationNotFound, VPCAssociationAuthorizationNotFound$1);
var VPCAssociationNotFound = [
    -3,
    n0,
    _VPCANF,
    {
        [_e]: _c,
        [_hE]: 404,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(VPCAssociationNotFound, VPCAssociationNotFound$1);
var Route53ServiceException = [-3, _sm, "Route53ServiceException", 0, [], []];
schema.TypeRegistry.for(_sm).registerError(Route53ServiceException, Route53ServiceException$1);
var Changes = [
    1,
    n0,
    _Cha,
    0,
    [
        () => Change,
        {
            [_xN]: _Ch,
        },
    ],
];
var ChildHealthCheckList = [
    1,
    n0,
    _CHCL,
    0,
    [
        0,
        {
            [_xN]: _CHCh,
        },
    ],
];
var CidrBlockSummaries = [1, n0, _CBSi, 0, () => CidrBlockSummary];
var CidrCollectionChanges = [1, n0, _CCCi, 0, [() => CidrCollectionChange, 0]];
var CidrList = [
    1,
    n0,
    _CL,
    0,
    [
        0,
        {
            [_xN]: _Ci,
        },
    ],
];
var CollectionSummaries = [1, n0, _CSo, 0, () => CollectionSummary];
var DelegationSetNameServers = [
    1,
    n0,
    _DSNS,
    0,
    [
        0,
        {
            [_xN]: _NSa,
        },
    ],
];
var DelegationSets = [
    1,
    n0,
    _DSe,
    0,
    [
        () => DelegationSet,
        {
            [_xN]: _DS,
        },
    ],
];
var DimensionList = [
    1,
    n0,
    _DL,
    0,
    [
        () => Dimension,
        {
            [_xN]: _Di,
        },
    ],
];
var ErrorMessages = [
    1,
    n0,
    _EM,
    0,
    [
        0,
        {
            [_xN]: _M,
        },
    ],
];
var GeoLocationDetailsList = [
    1,
    n0,
    _GLDL,
    0,
    [
        () => GeoLocationDetails,
        {
            [_xN]: _GLD,
        },
    ],
];
var HealthCheckObservations = [
    1,
    n0,
    _HCO,
    0,
    [
        () => HealthCheckObservation,
        {
            [_xN]: _HCOe,
        },
    ],
];
var HealthCheckRegionList = [
    1,
    n0,
    _HCRL,
    0,
    [
        0,
        {
            [_xN]: _R,
        },
    ],
];
var HealthChecks = [
    1,
    n0,
    _HCe,
    0,
    [
        () => HealthCheck,
        {
            [_xN]: _HC,
        },
    ],
];
var HostedZones = [
    1,
    n0,
    _HZo,
    0,
    [
        () => HostedZone,
        {
            [_xN]: _HZ,
        },
    ],
];
var HostedZoneSummaries = [
    1,
    n0,
    _HZSo,
    0,
    [
        () => HostedZoneSummary,
        {
            [_xN]: _HZS,
        },
    ],
];
var KeySigningKeys = [1, n0, _KSKe, 0, () => KeySigningKey];
var LocationSummaries = [1, n0, _LSoc, 0, () => LocationSummary];
var QueryLoggingConfigs = [
    1,
    n0,
    _QLCu,
    0,
    [
        () => QueryLoggingConfig,
        {
            [_xN]: _QLC,
        },
    ],
];
var RecordData = [
    1,
    n0,
    _RD,
    0,
    [
        0,
        {
            [_xN]: _RDE,
        },
    ],
];
var ResettableElementNameList = [
    1,
    n0,
    _RENL,
    0,
    [
        0,
        {
            [_xN]: _REN,
        },
    ],
];
var ResourceRecords = [
    1,
    n0,
    _RRe,
    0,
    [
        () => ResourceRecord,
        {
            [_xN]: _RR,
        },
    ],
];
var ResourceRecordSets = [
    1,
    n0,
    _RRSe,
    0,
    [
        () => ResourceRecordSet,
        {
            [_xN]: _RRS,
        },
    ],
];
var ResourceTagSetList = [
    1,
    n0,
    _RTSL,
    0,
    [
        () => ResourceTagSet,
        {
            [_xN]: _RTS,
        },
    ],
];
var TagKeyList = [
    1,
    n0,
    _TKL,
    0,
    [
        0,
        {
            [_xN]: _K,
        },
    ],
];
var TagList = [
    1,
    n0,
    _TL,
    0,
    [
        () => Tag,
        {
            [_xN]: _Tag,
        },
    ],
];
var TagResourceIdList = [
    1,
    n0,
    _TRIL,
    0,
    [
        0,
        {
            [_xN]: _RI,
        },
    ],
];
var TrafficPolicies = [
    1,
    n0,
    _TPr,
    0,
    [
        () => TrafficPolicy,
        {
            [_xN]: _TP,
        },
    ],
];
var TrafficPolicyInstances = [
    1,
    n0,
    _TPIra,
    0,
    [
        () => TrafficPolicyInstance,
        {
            [_xN]: _TPIr,
        },
    ],
];
var TrafficPolicySummaries = [
    1,
    n0,
    _TPS,
    0,
    [
        () => TrafficPolicySummary,
        {
            [_xN]: _TPSr,
        },
    ],
];
var VPCs = [
    1,
    n0,
    _VPCs,
    0,
    [
        () => VPC,
        {
            [_xN]: _VPC,
        },
    ],
];
var ActivateKeySigningKey = [
    9,
    n0,
    _AKSK,
    {
        [_ht]: ["POST", "/2013-04-01/keysigningkey/{HostedZoneId}/{Name}/activate", 200],
    },
    () => ActivateKeySigningKeyRequest,
    () => ActivateKeySigningKeyResponse,
];
var AssociateVPCWithHostedZone = [
    9,
    n0,
    _AVPCWHZ,
    {
        [_ht]: ["POST", "/2013-04-01/hostedzone/{HostedZoneId}/associatevpc", 200],
    },
    () => AssociateVPCWithHostedZoneRequest,
    () => AssociateVPCWithHostedZoneResponse,
];
var ChangeCidrCollection = [
    9,
    n0,
    _CCCh,
    {
        [_ht]: ["POST", "/2013-04-01/cidrcollection/{Id}", 200],
    },
    () => ChangeCidrCollectionRequest,
    () => ChangeCidrCollectionResponse,
];
var ChangeResourceRecordSets = [
    9,
    n0,
    _CRRS,
    {
        [_ht]: ["POST", "/2013-04-01/hostedzone/{HostedZoneId}/rrset", 200],
    },
    () => ChangeResourceRecordSetsRequest,
    () => ChangeResourceRecordSetsResponse,
];
var ChangeTagsForResource = [
    9,
    n0,
    _CTFR,
    {
        [_ht]: ["POST", "/2013-04-01/tags/{ResourceType}/{ResourceId}", 200],
    },
    () => ChangeTagsForResourceRequest,
    () => ChangeTagsForResourceResponse,
];
var CreateCidrCollection = [
    9,
    n0,
    _CCCr,
    {
        [_ht]: ["POST", "/2013-04-01/cidrcollection", 201],
    },
    () => CreateCidrCollectionRequest,
    () => CreateCidrCollectionResponse,
];
var CreateHealthCheck = [
    9,
    n0,
    _CHCr,
    {
        [_ht]: ["POST", "/2013-04-01/healthcheck", 201],
    },
    () => CreateHealthCheckRequest,
    () => CreateHealthCheckResponse,
];
var CreateHostedZone = [
    9,
    n0,
    _CHZ,
    {
        [_ht]: ["POST", "/2013-04-01/hostedzone", 201],
    },
    () => CreateHostedZoneRequest,
    () => CreateHostedZoneResponse,
];
var CreateKeySigningKey = [
    9,
    n0,
    _CKSK,
    {
        [_ht]: ["POST", "/2013-04-01/keysigningkey", 201],
    },
    () => CreateKeySigningKeyRequest,
    () => CreateKeySigningKeyResponse,
];
var CreateQueryLoggingConfig = [
    9,
    n0,
    _CQLC,
    {
        [_ht]: ["POST", "/2013-04-01/queryloggingconfig", 201],
    },
    () => CreateQueryLoggingConfigRequest,
    () => CreateQueryLoggingConfigResponse,
];
var CreateReusableDelegationSet = [
    9,
    n0,
    _CRDS,
    {
        [_ht]: ["POST", "/2013-04-01/delegationset", 201],
    },
    () => CreateReusableDelegationSetRequest,
    () => CreateReusableDelegationSetResponse,
];
var CreateTrafficPolicy = [
    9,
    n0,
    _CTP,
    {
        [_ht]: ["POST", "/2013-04-01/trafficpolicy", 201],
    },
    () => CreateTrafficPolicyRequest,
    () => CreateTrafficPolicyResponse,
];
var CreateTrafficPolicyInstance = [
    9,
    n0,
    _CTPI,
    {
        [_ht]: ["POST", "/2013-04-01/trafficpolicyinstance", 201],
    },
    () => CreateTrafficPolicyInstanceRequest,
    () => CreateTrafficPolicyInstanceResponse,
];
var CreateTrafficPolicyVersion = [
    9,
    n0,
    _CTPV,
    {
        [_ht]: ["POST", "/2013-04-01/trafficpolicy/{Id}", 201],
    },
    () => CreateTrafficPolicyVersionRequest,
    () => CreateTrafficPolicyVersionResponse,
];
var CreateVPCAssociationAuthorization = [
    9,
    n0,
    _CVPCAA,
    {
        [_ht]: ["POST", "/2013-04-01/hostedzone/{HostedZoneId}/authorizevpcassociation", 200],
    },
    () => CreateVPCAssociationAuthorizationRequest,
    () => CreateVPCAssociationAuthorizationResponse,
];
var DeactivateKeySigningKey = [
    9,
    n0,
    _DKSK,
    {
        [_ht]: ["POST", "/2013-04-01/keysigningkey/{HostedZoneId}/{Name}/deactivate", 200],
    },
    () => DeactivateKeySigningKeyRequest,
    () => DeactivateKeySigningKeyResponse,
];
var DeleteCidrCollection = [
    9,
    n0,
    _DCC,
    {
        [_ht]: ["DELETE", "/2013-04-01/cidrcollection/{Id}", 200],
    },
    () => DeleteCidrCollectionRequest,
    () => DeleteCidrCollectionResponse,
];
var DeleteHealthCheck = [
    9,
    n0,
    _DHC,
    {
        [_ht]: ["DELETE", "/2013-04-01/healthcheck/{HealthCheckId}", 200],
    },
    () => DeleteHealthCheckRequest,
    () => DeleteHealthCheckResponse,
];
var DeleteHostedZone = [
    9,
    n0,
    _DHZ,
    {
        [_ht]: ["DELETE", "/2013-04-01/hostedzone/{Id}", 200],
    },
    () => DeleteHostedZoneRequest,
    () => DeleteHostedZoneResponse,
];
var DeleteKeySigningKey = [
    9,
    n0,
    _DKSKe,
    {
        [_ht]: ["DELETE", "/2013-04-01/keysigningkey/{HostedZoneId}/{Name}", 200],
    },
    () => DeleteKeySigningKeyRequest,
    () => DeleteKeySigningKeyResponse,
];
var DeleteQueryLoggingConfig = [
    9,
    n0,
    _DQLC,
    {
        [_ht]: ["DELETE", "/2013-04-01/queryloggingconfig/{Id}", 200],
    },
    () => DeleteQueryLoggingConfigRequest,
    () => DeleteQueryLoggingConfigResponse,
];
var DeleteReusableDelegationSet = [
    9,
    n0,
    _DRDS,
    {
        [_ht]: ["DELETE", "/2013-04-01/delegationset/{Id}", 200],
    },
    () => DeleteReusableDelegationSetRequest,
    () => DeleteReusableDelegationSetResponse,
];
var DeleteTrafficPolicy = [
    9,
    n0,
    _DTP,
    {
        [_ht]: ["DELETE", "/2013-04-01/trafficpolicy/{Id}/{Version}", 200],
    },
    () => DeleteTrafficPolicyRequest,
    () => DeleteTrafficPolicyResponse,
];
var DeleteTrafficPolicyInstance = [
    9,
    n0,
    _DTPI,
    {
        [_ht]: ["DELETE", "/2013-04-01/trafficpolicyinstance/{Id}", 200],
    },
    () => DeleteTrafficPolicyInstanceRequest,
    () => DeleteTrafficPolicyInstanceResponse,
];
var DeleteVPCAssociationAuthorization = [
    9,
    n0,
    _DVPCAA,
    {
        [_ht]: ["POST", "/2013-04-01/hostedzone/{HostedZoneId}/deauthorizevpcassociation", 200],
    },
    () => DeleteVPCAssociationAuthorizationRequest,
    () => DeleteVPCAssociationAuthorizationResponse,
];
var DisableHostedZoneDNSSEC = [
    9,
    n0,
    _DHZDNSSEC,
    {
        [_ht]: ["POST", "/2013-04-01/hostedzone/{HostedZoneId}/disable-dnssec", 200],
    },
    () => DisableHostedZoneDNSSECRequest,
    () => DisableHostedZoneDNSSECResponse,
];
var DisassociateVPCFromHostedZone = [
    9,
    n0,
    _DVPCFHZ,
    {
        [_ht]: ["POST", "/2013-04-01/hostedzone/{HostedZoneId}/disassociatevpc", 200],
    },
    () => DisassociateVPCFromHostedZoneRequest,
    () => DisassociateVPCFromHostedZoneResponse,
];
var EnableHostedZoneDNSSEC = [
    9,
    n0,
    _EHZDNSSEC,
    {
        [_ht]: ["POST", "/2013-04-01/hostedzone/{HostedZoneId}/enable-dnssec", 200],
    },
    () => EnableHostedZoneDNSSECRequest,
    () => EnableHostedZoneDNSSECResponse,
];
var GetAccountLimit = [
    9,
    n0,
    _GAL,
    {
        [_ht]: ["GET", "/2013-04-01/accountlimit/{Type}", 200],
    },
    () => GetAccountLimitRequest,
    () => GetAccountLimitResponse,
];
var GetChange = [
    9,
    n0,
    _GC,
    {
        [_ht]: ["GET", "/2013-04-01/change/{Id}", 200],
    },
    () => GetChangeRequest,
    () => GetChangeResponse,
];
var GetCheckerIpRanges = [
    9,
    n0,
    _GCIR,
    {
        [_ht]: ["GET", "/2013-04-01/checkeripranges", 200],
    },
    () => GetCheckerIpRangesRequest,
    () => GetCheckerIpRangesResponse,
];
var GetDNSSEC = [
    9,
    n0,
    _GDNSSEC,
    {
        [_ht]: ["GET", "/2013-04-01/hostedzone/{HostedZoneId}/dnssec", 200],
    },
    () => GetDNSSECRequest,
    () => GetDNSSECResponse,
];
var GetGeoLocation = [
    9,
    n0,
    _GGL,
    {
        [_ht]: ["GET", "/2013-04-01/geolocation", 200],
    },
    () => GetGeoLocationRequest,
    () => GetGeoLocationResponse,
];
var GetHealthCheck = [
    9,
    n0,
    _GHC,
    {
        [_ht]: ["GET", "/2013-04-01/healthcheck/{HealthCheckId}", 200],
    },
    () => GetHealthCheckRequest,
    () => GetHealthCheckResponse,
];
var GetHealthCheckCount = [
    9,
    n0,
    _GHCC,
    {
        [_ht]: ["GET", "/2013-04-01/healthcheckcount", 200],
    },
    () => GetHealthCheckCountRequest,
    () => GetHealthCheckCountResponse,
];
var GetHealthCheckLastFailureReason = [
    9,
    n0,
    _GHCLFR,
    {
        [_ht]: ["GET", "/2013-04-01/healthcheck/{HealthCheckId}/lastfailurereason", 200],
    },
    () => GetHealthCheckLastFailureReasonRequest,
    () => GetHealthCheckLastFailureReasonResponse,
];
var GetHealthCheckStatus = [
    9,
    n0,
    _GHCS,
    {
        [_ht]: ["GET", "/2013-04-01/healthcheck/{HealthCheckId}/status", 200],
    },
    () => GetHealthCheckStatusRequest,
    () => GetHealthCheckStatusResponse,
];
var GetHostedZone = [
    9,
    n0,
    _GHZ,
    {
        [_ht]: ["GET", "/2013-04-01/hostedzone/{Id}", 200],
    },
    () => GetHostedZoneRequest,
    () => GetHostedZoneResponse,
];
var GetHostedZoneCount = [
    9,
    n0,
    _GHZC,
    {
        [_ht]: ["GET", "/2013-04-01/hostedzonecount", 200],
    },
    () => GetHostedZoneCountRequest,
    () => GetHostedZoneCountResponse,
];
var GetHostedZoneLimit = [
    9,
    n0,
    _GHZL,
    {
        [_ht]: ["GET", "/2013-04-01/hostedzonelimit/{HostedZoneId}/{Type}", 200],
    },
    () => GetHostedZoneLimitRequest,
    () => GetHostedZoneLimitResponse,
];
var GetQueryLoggingConfig = [
    9,
    n0,
    _GQLC,
    {
        [_ht]: ["GET", "/2013-04-01/queryloggingconfig/{Id}", 200],
    },
    () => GetQueryLoggingConfigRequest,
    () => GetQueryLoggingConfigResponse,
];
var GetReusableDelegationSet = [
    9,
    n0,
    _GRDS,
    {
        [_ht]: ["GET", "/2013-04-01/delegationset/{Id}", 200],
    },
    () => GetReusableDelegationSetRequest,
    () => GetReusableDelegationSetResponse,
];
var GetReusableDelegationSetLimit = [
    9,
    n0,
    _GRDSL,
    {
        [_ht]: ["GET", "/2013-04-01/reusabledelegationsetlimit/{DelegationSetId}/{Type}", 200],
    },
    () => GetReusableDelegationSetLimitRequest,
    () => GetReusableDelegationSetLimitResponse,
];
var GetTrafficPolicy = [
    9,
    n0,
    _GTP,
    {
        [_ht]: ["GET", "/2013-04-01/trafficpolicy/{Id}/{Version}", 200],
    },
    () => GetTrafficPolicyRequest,
    () => GetTrafficPolicyResponse,
];
var GetTrafficPolicyInstance = [
    9,
    n0,
    _GTPI,
    {
        [_ht]: ["GET", "/2013-04-01/trafficpolicyinstance/{Id}", 200],
    },
    () => GetTrafficPolicyInstanceRequest,
    () => GetTrafficPolicyInstanceResponse,
];
var GetTrafficPolicyInstanceCount = [
    9,
    n0,
    _GTPIC,
    {
        [_ht]: ["GET", "/2013-04-01/trafficpolicyinstancecount", 200],
    },
    () => GetTrafficPolicyInstanceCountRequest,
    () => GetTrafficPolicyInstanceCountResponse,
];
var ListCidrBlocks = [
    9,
    n0,
    _LCB,
    {
        [_ht]: ["GET", "/2013-04-01/cidrcollection/{CollectionId}/cidrblocks", 200],
    },
    () => ListCidrBlocksRequest,
    () => ListCidrBlocksResponse,
];
var ListCidrCollections = [
    9,
    n0,
    _LCC,
    {
        [_ht]: ["GET", "/2013-04-01/cidrcollection", 200],
    },
    () => ListCidrCollectionsRequest,
    () => ListCidrCollectionsResponse,
];
var ListCidrLocations = [
    9,
    n0,
    _LCL,
    {
        [_ht]: ["GET", "/2013-04-01/cidrcollection/{CollectionId}", 200],
    },
    () => ListCidrLocationsRequest,
    () => ListCidrLocationsResponse,
];
var ListGeoLocations = [
    9,
    n0,
    _LGL,
    {
        [_ht]: ["GET", "/2013-04-01/geolocations", 200],
    },
    () => ListGeoLocationsRequest,
    () => ListGeoLocationsResponse,
];
var ListHealthChecks = [
    9,
    n0,
    _LHC,
    {
        [_ht]: ["GET", "/2013-04-01/healthcheck", 200],
    },
    () => ListHealthChecksRequest,
    () => ListHealthChecksResponse,
];
var ListHostedZones = [
    9,
    n0,
    _LHZ,
    {
        [_ht]: ["GET", "/2013-04-01/hostedzone", 200],
    },
    () => ListHostedZonesRequest,
    () => ListHostedZonesResponse,
];
var ListHostedZonesByName = [
    9,
    n0,
    _LHZBN,
    {
        [_ht]: ["GET", "/2013-04-01/hostedzonesbyname", 200],
    },
    () => ListHostedZonesByNameRequest,
    () => ListHostedZonesByNameResponse,
];
var ListHostedZonesByVPC = [
    9,
    n0,
    _LHZBVPC,
    {
        [_ht]: ["GET", "/2013-04-01/hostedzonesbyvpc", 200],
    },
    () => ListHostedZonesByVPCRequest,
    () => ListHostedZonesByVPCResponse,
];
var ListQueryLoggingConfigs = [
    9,
    n0,
    _LQLC,
    {
        [_ht]: ["GET", "/2013-04-01/queryloggingconfig", 200],
    },
    () => ListQueryLoggingConfigsRequest,
    () => ListQueryLoggingConfigsResponse,
];
var ListResourceRecordSets = [
    9,
    n0,
    _LRRS,
    {
        [_ht]: ["GET", "/2013-04-01/hostedzone/{HostedZoneId}/rrset", 200],
    },
    () => ListResourceRecordSetsRequest,
    () => ListResourceRecordSetsResponse,
];
var ListReusableDelegationSets = [
    9,
    n0,
    _LRDS,
    {
        [_ht]: ["GET", "/2013-04-01/delegationset", 200],
    },
    () => ListReusableDelegationSetsRequest,
    () => ListReusableDelegationSetsResponse,
];
var ListTagsForResource = [
    9,
    n0,
    _LTFR,
    {
        [_ht]: ["GET", "/2013-04-01/tags/{ResourceType}/{ResourceId}", 200],
    },
    () => ListTagsForResourceRequest,
    () => ListTagsForResourceResponse,
];
var ListTagsForResources = [
    9,
    n0,
    _LTFRi,
    {
        [_ht]: ["POST", "/2013-04-01/tags/{ResourceType}", 200],
    },
    () => ListTagsForResourcesRequest,
    () => ListTagsForResourcesResponse,
];
var ListTrafficPolicies = [
    9,
    n0,
    _LTP,
    {
        [_ht]: ["GET", "/2013-04-01/trafficpolicies", 200],
    },
    () => ListTrafficPoliciesRequest,
    () => ListTrafficPoliciesResponse,
];
var ListTrafficPolicyInstances = [
    9,
    n0,
    _LTPI,
    {
        [_ht]: ["GET", "/2013-04-01/trafficpolicyinstances", 200],
    },
    () => ListTrafficPolicyInstancesRequest,
    () => ListTrafficPolicyInstancesResponse,
];
var ListTrafficPolicyInstancesByHostedZone = [
    9,
    n0,
    _LTPIBHZ,
    {
        [_ht]: ["GET", "/2013-04-01/trafficpolicyinstances/hostedzone", 200],
    },
    () => ListTrafficPolicyInstancesByHostedZoneRequest,
    () => ListTrafficPolicyInstancesByHostedZoneResponse,
];
var ListTrafficPolicyInstancesByPolicy = [
    9,
    n0,
    _LTPIBP,
    {
        [_ht]: ["GET", "/2013-04-01/trafficpolicyinstances/trafficpolicy", 200],
    },
    () => ListTrafficPolicyInstancesByPolicyRequest,
    () => ListTrafficPolicyInstancesByPolicyResponse,
];
var ListTrafficPolicyVersions = [
    9,
    n0,
    _LTPV,
    {
        [_ht]: ["GET", "/2013-04-01/trafficpolicies/{Id}/versions", 200],
    },
    () => ListTrafficPolicyVersionsRequest,
    () => ListTrafficPolicyVersionsResponse,
];
var ListVPCAssociationAuthorizations = [
    9,
    n0,
    _LVPCAA,
    {
        [_ht]: ["GET", "/2013-04-01/hostedzone/{HostedZoneId}/authorizevpcassociation", 200],
    },
    () => ListVPCAssociationAuthorizationsRequest,
    () => ListVPCAssociationAuthorizationsResponse,
];
var TestDNSAnswer = [
    9,
    n0,
    _TDNSA,
    {
        [_ht]: ["GET", "/2013-04-01/testdnsanswer", 200],
    },
    () => TestDNSAnswerRequest,
    () => TestDNSAnswerResponse,
];
var UpdateHealthCheck = [
    9,
    n0,
    _UHC,
    {
        [_ht]: ["POST", "/2013-04-01/healthcheck/{HealthCheckId}", 200],
    },
    () => UpdateHealthCheckRequest,
    () => UpdateHealthCheckResponse,
];
var UpdateHostedZoneComment = [
    9,
    n0,
    _UHZC,
    {
        [_ht]: ["POST", "/2013-04-01/hostedzone/{Id}", 200],
    },
    () => UpdateHostedZoneCommentRequest,
    () => UpdateHostedZoneCommentResponse,
];
var UpdateTrafficPolicyComment = [
    9,
    n0,
    _UTPC,
    {
        [_ht]: ["POST", "/2013-04-01/trafficpolicy/{Id}/{Version}", 200],
    },
    () => UpdateTrafficPolicyCommentRequest,
    () => UpdateTrafficPolicyCommentResponse,
];
var UpdateTrafficPolicyInstance = [
    9,
    n0,
    _UTPI,
    {
        [_ht]: ["POST", "/2013-04-01/trafficpolicyinstance/{Id}", 200],
    },
    () => UpdateTrafficPolicyInstanceRequest,
    () => UpdateTrafficPolicyInstanceResponse,
];

class ActivateKeySigningKeyCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions()), middlewareSdkRoute53.getIdNormalizerPlugin(config)];
})
    .s("AWSDnsV20130401", "ActivateKeySigningKey", {})
    .n("Route53Client", "ActivateKeySigningKeyCommand")
    .sc(ActivateKeySigningKey)
    .build() {
}

class AssociateVPCWithHostedZoneCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions()), middlewareSdkRoute53.getIdNormalizerPlugin(config)];
})
    .s("AWSDnsV20130401", "AssociateVPCWithHostedZone", {})
    .n("Route53Client", "AssociateVPCWithHostedZoneCommand")
    .sc(AssociateVPCWithHostedZone)
    .build() {
}

class ChangeCidrCollectionCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions()), middlewareSdkRoute53.getIdNormalizerPlugin(config)];
})
    .s("AWSDnsV20130401", "ChangeCidrCollection", {})
    .n("Route53Client", "ChangeCidrCollectionCommand")
    .sc(ChangeCidrCollection)
    .build() {
}

class ChangeResourceRecordSetsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [
        middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions()),
        middlewareSdkRoute53.getChangeResourceRecordSetsPlugin(config),
        middlewareSdkRoute53.getIdNormalizerPlugin(config),
    ];
})
    .s("AWSDnsV20130401", "ChangeResourceRecordSets", {})
    .n("Route53Client", "ChangeResourceRecordSetsCommand")
    .sc(ChangeResourceRecordSets)
    .build() {
}

class ChangeTagsForResourceCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSDnsV20130401", "ChangeTagsForResource", {})
    .n("Route53Client", "ChangeTagsForResourceCommand")
    .sc(ChangeTagsForResource)
    .build() {
}

class CreateCidrCollectionCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSDnsV20130401", "CreateCidrCollection", {})
    .n("Route53Client", "CreateCidrCollectionCommand")
    .sc(CreateCidrCollection)
    .build() {
}

class CreateHealthCheckCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSDnsV20130401", "CreateHealthCheck", {})
    .n("Route53Client", "CreateHealthCheckCommand")
    .sc(CreateHealthCheck)
    .build() {
}

class CreateHostedZoneCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions()), middlewareSdkRoute53.getIdNormalizerPlugin(config)];
})
    .s("AWSDnsV20130401", "CreateHostedZone", {})
    .n("Route53Client", "CreateHostedZoneCommand")
    .sc(CreateHostedZone)
    .build() {
}

class CreateKeySigningKeyCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions()), middlewareSdkRoute53.getIdNormalizerPlugin(config)];
})
    .s("AWSDnsV20130401", "CreateKeySigningKey", {})
    .n("Route53Client", "CreateKeySigningKeyCommand")
    .sc(CreateKeySigningKey)
    .build() {
}

class CreateQueryLoggingConfigCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions()), middlewareSdkRoute53.getIdNormalizerPlugin(config)];
})
    .s("AWSDnsV20130401", "CreateQueryLoggingConfig", {})
    .n("Route53Client", "CreateQueryLoggingConfigCommand")
    .sc(CreateQueryLoggingConfig)
    .build() {
}

class CreateReusableDelegationSetCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions()), middlewareSdkRoute53.getIdNormalizerPlugin(config)];
})
    .s("AWSDnsV20130401", "CreateReusableDelegationSet", {})
    .n("Route53Client", "CreateReusableDelegationSetCommand")
    .sc(CreateReusableDelegationSet)
    .build() {
}

class CreateTrafficPolicyCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSDnsV20130401", "CreateTrafficPolicy", {})
    .n("Route53Client", "CreateTrafficPolicyCommand")
    .sc(CreateTrafficPolicy)
    .build() {
}

class CreateTrafficPolicyInstanceCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions()), middlewareSdkRoute53.getIdNormalizerPlugin(config)];
})
    .s("AWSDnsV20130401", "CreateTrafficPolicyInstance", {})
    .n("Route53Client", "CreateTrafficPolicyInstanceCommand")
    .sc(CreateTrafficPolicyInstance)
    .build() {
}

class CreateTrafficPolicyVersionCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions()), middlewareSdkRoute53.getIdNormalizerPlugin(config)];
})
    .s("AWSDnsV20130401", "CreateTrafficPolicyVersion", {})
    .n("Route53Client", "CreateTrafficPolicyVersionCommand")
    .sc(CreateTrafficPolicyVersion)
    .build() {
}

class CreateVPCAssociationAuthorizationCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions()), middlewareSdkRoute53.getIdNormalizerPlugin(config)];
})
    .s("AWSDnsV20130401", "CreateVPCAssociationAuthorization", {})
    .n("Route53Client", "CreateVPCAssociationAuthorizationCommand")
    .sc(CreateVPCAssociationAuthorization)
    .build() {
}

class DeactivateKeySigningKeyCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions()), middlewareSdkRoute53.getIdNormalizerPlugin(config)];
})
    .s("AWSDnsV20130401", "DeactivateKeySigningKey", {})
    .n("Route53Client", "DeactivateKeySigningKeyCommand")
    .sc(DeactivateKeySigningKey)
    .build() {
}

class DeleteCidrCollectionCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions()), middlewareSdkRoute53.getIdNormalizerPlugin(config)];
})
    .s("AWSDnsV20130401", "DeleteCidrCollection", {})
    .n("Route53Client", "DeleteCidrCollectionCommand")
    .sc(DeleteCidrCollection)
    .build() {
}

class DeleteHealthCheckCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSDnsV20130401", "DeleteHealthCheck", {})
    .n("Route53Client", "DeleteHealthCheckCommand")
    .sc(DeleteHealthCheck)
    .build() {
}

class DeleteHostedZoneCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions()), middlewareSdkRoute53.getIdNormalizerPlugin(config)];
})
    .s("AWSDnsV20130401", "DeleteHostedZone", {})
    .n("Route53Client", "DeleteHostedZoneCommand")
    .sc(DeleteHostedZone)
    .build() {
}

class DeleteKeySigningKeyCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions()), middlewareSdkRoute53.getIdNormalizerPlugin(config)];
})
    .s("AWSDnsV20130401", "DeleteKeySigningKey", {})
    .n("Route53Client", "DeleteKeySigningKeyCommand")
    .sc(DeleteKeySigningKey)
    .build() {
}

class DeleteQueryLoggingConfigCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions()), middlewareSdkRoute53.getIdNormalizerPlugin(config)];
})
    .s("AWSDnsV20130401", "DeleteQueryLoggingConfig", {})
    .n("Route53Client", "DeleteQueryLoggingConfigCommand")
    .sc(DeleteQueryLoggingConfig)
    .build() {
}

class DeleteReusableDelegationSetCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions()), middlewareSdkRoute53.getIdNormalizerPlugin(config)];
})
    .s("AWSDnsV20130401", "DeleteReusableDelegationSet", {})
    .n("Route53Client", "DeleteReusableDelegationSetCommand")
    .sc(DeleteReusableDelegationSet)
    .build() {
}

class DeleteTrafficPolicyCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions()), middlewareSdkRoute53.getIdNormalizerPlugin(config)];
})
    .s("AWSDnsV20130401", "DeleteTrafficPolicy", {})
    .n("Route53Client", "DeleteTrafficPolicyCommand")
    .sc(DeleteTrafficPolicy)
    .build() {
}

class DeleteTrafficPolicyInstanceCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions()), middlewareSdkRoute53.getIdNormalizerPlugin(config)];
})
    .s("AWSDnsV20130401", "DeleteTrafficPolicyInstance", {})
    .n("Route53Client", "DeleteTrafficPolicyInstanceCommand")
    .sc(DeleteTrafficPolicyInstance)
    .build() {
}

class DeleteVPCAssociationAuthorizationCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions()), middlewareSdkRoute53.getIdNormalizerPlugin(config)];
})
    .s("AWSDnsV20130401", "DeleteVPCAssociationAuthorization", {})
    .n("Route53Client", "DeleteVPCAssociationAuthorizationCommand")
    .sc(DeleteVPCAssociationAuthorization)
    .build() {
}

class DisableHostedZoneDNSSECCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions()), middlewareSdkRoute53.getIdNormalizerPlugin(config)];
})
    .s("AWSDnsV20130401", "DisableHostedZoneDNSSEC", {})
    .n("Route53Client", "DisableHostedZoneDNSSECCommand")
    .sc(DisableHostedZoneDNSSEC)
    .build() {
}

class DisassociateVPCFromHostedZoneCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions()), middlewareSdkRoute53.getIdNormalizerPlugin(config)];
})
    .s("AWSDnsV20130401", "DisassociateVPCFromHostedZone", {})
    .n("Route53Client", "DisassociateVPCFromHostedZoneCommand")
    .sc(DisassociateVPCFromHostedZone)
    .build() {
}

class EnableHostedZoneDNSSECCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions()), middlewareSdkRoute53.getIdNormalizerPlugin(config)];
})
    .s("AWSDnsV20130401", "EnableHostedZoneDNSSEC", {})
    .n("Route53Client", "EnableHostedZoneDNSSECCommand")
    .sc(EnableHostedZoneDNSSEC)
    .build() {
}

class GetAccountLimitCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSDnsV20130401", "GetAccountLimit", {})
    .n("Route53Client", "GetAccountLimitCommand")
    .sc(GetAccountLimit)
    .build() {
}

class GetChangeCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions()), middlewareSdkRoute53.getIdNormalizerPlugin(config)];
})
    .s("AWSDnsV20130401", "GetChange", {})
    .n("Route53Client", "GetChangeCommand")
    .sc(GetChange)
    .build() {
}

class GetCheckerIpRangesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSDnsV20130401", "GetCheckerIpRanges", {})
    .n("Route53Client", "GetCheckerIpRangesCommand")
    .sc(GetCheckerIpRanges)
    .build() {
}

class GetDNSSECCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions()), middlewareSdkRoute53.getIdNormalizerPlugin(config)];
})
    .s("AWSDnsV20130401", "GetDNSSEC", {})
    .n("Route53Client", "GetDNSSECCommand")
    .sc(GetDNSSEC)
    .build() {
}

class GetGeoLocationCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSDnsV20130401", "GetGeoLocation", {})
    .n("Route53Client", "GetGeoLocationCommand")
    .sc(GetGeoLocation)
    .build() {
}

class GetHealthCheckCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSDnsV20130401", "GetHealthCheck", {})
    .n("Route53Client", "GetHealthCheckCommand")
    .sc(GetHealthCheck)
    .build() {
}

class GetHealthCheckCountCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSDnsV20130401", "GetHealthCheckCount", {})
    .n("Route53Client", "GetHealthCheckCountCommand")
    .sc(GetHealthCheckCount)
    .build() {
}

class GetHealthCheckLastFailureReasonCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSDnsV20130401", "GetHealthCheckLastFailureReason", {})
    .n("Route53Client", "GetHealthCheckLastFailureReasonCommand")
    .sc(GetHealthCheckLastFailureReason)
    .build() {
}

class GetHealthCheckStatusCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSDnsV20130401", "GetHealthCheckStatus", {})
    .n("Route53Client", "GetHealthCheckStatusCommand")
    .sc(GetHealthCheckStatus)
    .build() {
}

class GetHostedZoneCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions()), middlewareSdkRoute53.getIdNormalizerPlugin(config)];
})
    .s("AWSDnsV20130401", "GetHostedZone", {})
    .n("Route53Client", "GetHostedZoneCommand")
    .sc(GetHostedZone)
    .build() {
}

class GetHostedZoneCountCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSDnsV20130401", "GetHostedZoneCount", {})
    .n("Route53Client", "GetHostedZoneCountCommand")
    .sc(GetHostedZoneCount)
    .build() {
}

class GetHostedZoneLimitCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions()), middlewareSdkRoute53.getIdNormalizerPlugin(config)];
})
    .s("AWSDnsV20130401", "GetHostedZoneLimit", {})
    .n("Route53Client", "GetHostedZoneLimitCommand")
    .sc(GetHostedZoneLimit)
    .build() {
}

class GetQueryLoggingConfigCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions()), middlewareSdkRoute53.getIdNormalizerPlugin(config)];
})
    .s("AWSDnsV20130401", "GetQueryLoggingConfig", {})
    .n("Route53Client", "GetQueryLoggingConfigCommand")
    .sc(GetQueryLoggingConfig)
    .build() {
}

class GetReusableDelegationSetCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions()), middlewareSdkRoute53.getIdNormalizerPlugin(config)];
})
    .s("AWSDnsV20130401", "GetReusableDelegationSet", {})
    .n("Route53Client", "GetReusableDelegationSetCommand")
    .sc(GetReusableDelegationSet)
    .build() {
}

class GetReusableDelegationSetLimitCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions()), middlewareSdkRoute53.getIdNormalizerPlugin(config)];
})
    .s("AWSDnsV20130401", "GetReusableDelegationSetLimit", {})
    .n("Route53Client", "GetReusableDelegationSetLimitCommand")
    .sc(GetReusableDelegationSetLimit)
    .build() {
}

class GetTrafficPolicyCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions()), middlewareSdkRoute53.getIdNormalizerPlugin(config)];
})
    .s("AWSDnsV20130401", "GetTrafficPolicy", {})
    .n("Route53Client", "GetTrafficPolicyCommand")
    .sc(GetTrafficPolicy)
    .build() {
}

class GetTrafficPolicyInstanceCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions()), middlewareSdkRoute53.getIdNormalizerPlugin(config)];
})
    .s("AWSDnsV20130401", "GetTrafficPolicyInstance", {})
    .n("Route53Client", "GetTrafficPolicyInstanceCommand")
    .sc(GetTrafficPolicyInstance)
    .build() {
}

class GetTrafficPolicyInstanceCountCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSDnsV20130401", "GetTrafficPolicyInstanceCount", {})
    .n("Route53Client", "GetTrafficPolicyInstanceCountCommand")
    .sc(GetTrafficPolicyInstanceCount)
    .build() {
}

class ListCidrBlocksCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSDnsV20130401", "ListCidrBlocks", {})
    .n("Route53Client", "ListCidrBlocksCommand")
    .sc(ListCidrBlocks)
    .build() {
}

class ListCidrCollectionsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSDnsV20130401", "ListCidrCollections", {})
    .n("Route53Client", "ListCidrCollectionsCommand")
    .sc(ListCidrCollections)
    .build() {
}

class ListCidrLocationsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSDnsV20130401", "ListCidrLocations", {})
    .n("Route53Client", "ListCidrLocationsCommand")
    .sc(ListCidrLocations)
    .build() {
}

class ListGeoLocationsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSDnsV20130401", "ListGeoLocations", {})
    .n("Route53Client", "ListGeoLocationsCommand")
    .sc(ListGeoLocations)
    .build() {
}

class ListHealthChecksCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSDnsV20130401", "ListHealthChecks", {})
    .n("Route53Client", "ListHealthChecksCommand")
    .sc(ListHealthChecks)
    .build() {
}

class ListHostedZonesByNameCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions()), middlewareSdkRoute53.getIdNormalizerPlugin(config)];
})
    .s("AWSDnsV20130401", "ListHostedZonesByName", {})
    .n("Route53Client", "ListHostedZonesByNameCommand")
    .sc(ListHostedZonesByName)
    .build() {
}

class ListHostedZonesByVPCCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSDnsV20130401", "ListHostedZonesByVPC", {})
    .n("Route53Client", "ListHostedZonesByVPCCommand")
    .sc(ListHostedZonesByVPC)
    .build() {
}

class ListHostedZonesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions()), middlewareSdkRoute53.getIdNormalizerPlugin(config)];
})
    .s("AWSDnsV20130401", "ListHostedZones", {})
    .n("Route53Client", "ListHostedZonesCommand")
    .sc(ListHostedZones)
    .build() {
}

class ListQueryLoggingConfigsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions()), middlewareSdkRoute53.getIdNormalizerPlugin(config)];
})
    .s("AWSDnsV20130401", "ListQueryLoggingConfigs", {})
    .n("Route53Client", "ListQueryLoggingConfigsCommand")
    .sc(ListQueryLoggingConfigs)
    .build() {
}

class ListResourceRecordSetsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions()), middlewareSdkRoute53.getIdNormalizerPlugin(config)];
})
    .s("AWSDnsV20130401", "ListResourceRecordSets", {})
    .n("Route53Client", "ListResourceRecordSetsCommand")
    .sc(ListResourceRecordSets)
    .build() {
}

class ListReusableDelegationSetsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSDnsV20130401", "ListReusableDelegationSets", {})
    .n("Route53Client", "ListReusableDelegationSetsCommand")
    .sc(ListReusableDelegationSets)
    .build() {
}

class ListTagsForResourceCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSDnsV20130401", "ListTagsForResource", {})
    .n("Route53Client", "ListTagsForResourceCommand")
    .sc(ListTagsForResource)
    .build() {
}

class ListTagsForResourcesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSDnsV20130401", "ListTagsForResources", {})
    .n("Route53Client", "ListTagsForResourcesCommand")
    .sc(ListTagsForResources)
    .build() {
}

class ListTrafficPoliciesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSDnsV20130401", "ListTrafficPolicies", {})
    .n("Route53Client", "ListTrafficPoliciesCommand")
    .sc(ListTrafficPolicies)
    .build() {
}

class ListTrafficPolicyInstancesByHostedZoneCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions()), middlewareSdkRoute53.getIdNormalizerPlugin(config)];
})
    .s("AWSDnsV20130401", "ListTrafficPolicyInstancesByHostedZone", {})
    .n("Route53Client", "ListTrafficPolicyInstancesByHostedZoneCommand")
    .sc(ListTrafficPolicyInstancesByHostedZone)
    .build() {
}

class ListTrafficPolicyInstancesByPolicyCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSDnsV20130401", "ListTrafficPolicyInstancesByPolicy", {})
    .n("Route53Client", "ListTrafficPolicyInstancesByPolicyCommand")
    .sc(ListTrafficPolicyInstancesByPolicy)
    .build() {
}

class ListTrafficPolicyInstancesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSDnsV20130401", "ListTrafficPolicyInstances", {})
    .n("Route53Client", "ListTrafficPolicyInstancesCommand")
    .sc(ListTrafficPolicyInstances)
    .build() {
}

class ListTrafficPolicyVersionsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions()), middlewareSdkRoute53.getIdNormalizerPlugin(config)];
})
    .s("AWSDnsV20130401", "ListTrafficPolicyVersions", {})
    .n("Route53Client", "ListTrafficPolicyVersionsCommand")
    .sc(ListTrafficPolicyVersions)
    .build() {
}

class ListVPCAssociationAuthorizationsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions()), middlewareSdkRoute53.getIdNormalizerPlugin(config)];
})
    .s("AWSDnsV20130401", "ListVPCAssociationAuthorizations", {})
    .n("Route53Client", "ListVPCAssociationAuthorizationsCommand")
    .sc(ListVPCAssociationAuthorizations)
    .build() {
}

class TestDNSAnswerCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions()), middlewareSdkRoute53.getIdNormalizerPlugin(config)];
})
    .s("AWSDnsV20130401", "TestDNSAnswer", {})
    .n("Route53Client", "TestDNSAnswerCommand")
    .sc(TestDNSAnswer)
    .build() {
}

class UpdateHealthCheckCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSDnsV20130401", "UpdateHealthCheck", {})
    .n("Route53Client", "UpdateHealthCheckCommand")
    .sc(UpdateHealthCheck)
    .build() {
}

class UpdateHostedZoneCommentCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions()), middlewareSdkRoute53.getIdNormalizerPlugin(config)];
})
    .s("AWSDnsV20130401", "UpdateHostedZoneComment", {})
    .n("Route53Client", "UpdateHostedZoneCommentCommand")
    .sc(UpdateHostedZoneComment)
    .build() {
}

class UpdateTrafficPolicyCommentCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions()), middlewareSdkRoute53.getIdNormalizerPlugin(config)];
})
    .s("AWSDnsV20130401", "UpdateTrafficPolicyComment", {})
    .n("Route53Client", "UpdateTrafficPolicyCommentCommand")
    .sc(UpdateTrafficPolicyComment)
    .build() {
}

class UpdateTrafficPolicyInstanceCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions()), middlewareSdkRoute53.getIdNormalizerPlugin(config)];
})
    .s("AWSDnsV20130401", "UpdateTrafficPolicyInstance", {})
    .n("Route53Client", "UpdateTrafficPolicyInstanceCommand")
    .sc(UpdateTrafficPolicyInstance)
    .build() {
}

const commands = {
    ActivateKeySigningKeyCommand,
    AssociateVPCWithHostedZoneCommand,
    ChangeCidrCollectionCommand,
    ChangeResourceRecordSetsCommand,
    ChangeTagsForResourceCommand,
    CreateCidrCollectionCommand,
    CreateHealthCheckCommand,
    CreateHostedZoneCommand,
    CreateKeySigningKeyCommand,
    CreateQueryLoggingConfigCommand,
    CreateReusableDelegationSetCommand,
    CreateTrafficPolicyCommand,
    CreateTrafficPolicyInstanceCommand,
    CreateTrafficPolicyVersionCommand,
    CreateVPCAssociationAuthorizationCommand,
    DeactivateKeySigningKeyCommand,
    DeleteCidrCollectionCommand,
    DeleteHealthCheckCommand,
    DeleteHostedZoneCommand,
    DeleteKeySigningKeyCommand,
    DeleteQueryLoggingConfigCommand,
    DeleteReusableDelegationSetCommand,
    DeleteTrafficPolicyCommand,
    DeleteTrafficPolicyInstanceCommand,
    DeleteVPCAssociationAuthorizationCommand,
    DisableHostedZoneDNSSECCommand,
    DisassociateVPCFromHostedZoneCommand,
    EnableHostedZoneDNSSECCommand,
    GetAccountLimitCommand,
    GetChangeCommand,
    GetCheckerIpRangesCommand,
    GetDNSSECCommand,
    GetGeoLocationCommand,
    GetHealthCheckCommand,
    GetHealthCheckCountCommand,
    GetHealthCheckLastFailureReasonCommand,
    GetHealthCheckStatusCommand,
    GetHostedZoneCommand,
    GetHostedZoneCountCommand,
    GetHostedZoneLimitCommand,
    GetQueryLoggingConfigCommand,
    GetReusableDelegationSetCommand,
    GetReusableDelegationSetLimitCommand,
    GetTrafficPolicyCommand,
    GetTrafficPolicyInstanceCommand,
    GetTrafficPolicyInstanceCountCommand,
    ListCidrBlocksCommand,
    ListCidrCollectionsCommand,
    ListCidrLocationsCommand,
    ListGeoLocationsCommand,
    ListHealthChecksCommand,
    ListHostedZonesCommand,
    ListHostedZonesByNameCommand,
    ListHostedZonesByVPCCommand,
    ListQueryLoggingConfigsCommand,
    ListResourceRecordSetsCommand,
    ListReusableDelegationSetsCommand,
    ListTagsForResourceCommand,
    ListTagsForResourcesCommand,
    ListTrafficPoliciesCommand,
    ListTrafficPolicyInstancesCommand,
    ListTrafficPolicyInstancesByHostedZoneCommand,
    ListTrafficPolicyInstancesByPolicyCommand,
    ListTrafficPolicyVersionsCommand,
    ListVPCAssociationAuthorizationsCommand,
    TestDNSAnswerCommand,
    UpdateHealthCheckCommand,
    UpdateHostedZoneCommentCommand,
    UpdateTrafficPolicyCommentCommand,
    UpdateTrafficPolicyInstanceCommand,
};
class Route53 extends Route53Client {
}
smithyClient.createAggregatedClient(commands, Route53);

const paginateListCidrBlocks = core.createPaginator(Route53Client, ListCidrBlocksCommand, "NextToken", "NextToken", "MaxResults");

const paginateListCidrCollections = core.createPaginator(Route53Client, ListCidrCollectionsCommand, "NextToken", "NextToken", "MaxResults");

const paginateListCidrLocations = core.createPaginator(Route53Client, ListCidrLocationsCommand, "NextToken", "NextToken", "MaxResults");

const paginateListHealthChecks = core.createPaginator(Route53Client, ListHealthChecksCommand, "Marker", "NextMarker", "MaxItems");

const paginateListHostedZones = core.createPaginator(Route53Client, ListHostedZonesCommand, "Marker", "NextMarker", "MaxItems");

const paginateListQueryLoggingConfigs = core.createPaginator(Route53Client, ListQueryLoggingConfigsCommand, "NextToken", "NextToken", "MaxResults");

const checkState = async (client, input) => {
    let reason;
    try {
        const result = await client.send(new GetChangeCommand(input));
        reason = result;
        try {
            const returnComparator = () => {
                return result.ChangeInfo.Status;
            };
            if (returnComparator() === "INSYNC") {
                return { state: utilWaiter.WaiterState.SUCCESS, reason };
            }
        }
        catch (e) { }
    }
    catch (exception) {
        reason = exception;
    }
    return { state: utilWaiter.WaiterState.RETRY, reason };
};
const waitForResourceRecordSetsChanged = async (params, input) => {
    const serviceDefaults = { minDelay: 30, maxDelay: 120 };
    return utilWaiter.createWaiter({ ...serviceDefaults, ...params }, input, checkState);
};
const waitUntilResourceRecordSetsChanged = async (params, input) => {
    const serviceDefaults = { minDelay: 30, maxDelay: 120 };
    const result = await utilWaiter.createWaiter({ ...serviceDefaults, ...params }, input, checkState);
    return utilWaiter.checkExceptions(result);
};

Object.defineProperty(exports, "$Command", {
    enumerable: true,
    get: function () { return smithyClient.Command; }
});
Object.defineProperty(exports, "__Client", {
    enumerable: true,
    get: function () { return smithyClient.Client; }
});
exports.AccountLimitType = AccountLimitType;
exports.ActivateKeySigningKeyCommand = ActivateKeySigningKeyCommand;
exports.AssociateVPCWithHostedZoneCommand = AssociateVPCWithHostedZoneCommand;
exports.ChangeAction = ChangeAction;
exports.ChangeCidrCollectionCommand = ChangeCidrCollectionCommand;
exports.ChangeResourceRecordSetsCommand = ChangeResourceRecordSetsCommand;
exports.ChangeStatus = ChangeStatus;
exports.ChangeTagsForResourceCommand = ChangeTagsForResourceCommand;
exports.CidrBlockInUseException = CidrBlockInUseException$1;
exports.CidrCollectionAlreadyExistsException = CidrCollectionAlreadyExistsException$1;
exports.CidrCollectionChangeAction = CidrCollectionChangeAction;
exports.CidrCollectionInUseException = CidrCollectionInUseException$1;
exports.CidrCollectionVersionMismatchException = CidrCollectionVersionMismatchException$1;
exports.CloudWatchRegion = CloudWatchRegion;
exports.ComparisonOperator = ComparisonOperator;
exports.ConcurrentModification = ConcurrentModification$1;
exports.ConflictingDomainExists = ConflictingDomainExists$1;
exports.ConflictingTypes = ConflictingTypes$1;
exports.CreateCidrCollectionCommand = CreateCidrCollectionCommand;
exports.CreateHealthCheckCommand = CreateHealthCheckCommand;
exports.CreateHostedZoneCommand = CreateHostedZoneCommand;
exports.CreateKeySigningKeyCommand = CreateKeySigningKeyCommand;
exports.CreateQueryLoggingConfigCommand = CreateQueryLoggingConfigCommand;
exports.CreateReusableDelegationSetCommand = CreateReusableDelegationSetCommand;
exports.CreateTrafficPolicyCommand = CreateTrafficPolicyCommand;
exports.CreateTrafficPolicyInstanceCommand = CreateTrafficPolicyInstanceCommand;
exports.CreateTrafficPolicyVersionCommand = CreateTrafficPolicyVersionCommand;
exports.CreateVPCAssociationAuthorizationCommand = CreateVPCAssociationAuthorizationCommand;
exports.DNSSECNotFound = DNSSECNotFound$1;
exports.DeactivateKeySigningKeyCommand = DeactivateKeySigningKeyCommand;
exports.DelegationSetAlreadyCreated = DelegationSetAlreadyCreated$1;
exports.DelegationSetAlreadyReusable = DelegationSetAlreadyReusable$1;
exports.DelegationSetInUse = DelegationSetInUse$1;
exports.DelegationSetNotAvailable = DelegationSetNotAvailable$1;
exports.DelegationSetNotReusable = DelegationSetNotReusable$1;
exports.DeleteCidrCollectionCommand = DeleteCidrCollectionCommand;
exports.DeleteHealthCheckCommand = DeleteHealthCheckCommand;
exports.DeleteHostedZoneCommand = DeleteHostedZoneCommand;
exports.DeleteKeySigningKeyCommand = DeleteKeySigningKeyCommand;
exports.DeleteQueryLoggingConfigCommand = DeleteQueryLoggingConfigCommand;
exports.DeleteReusableDelegationSetCommand = DeleteReusableDelegationSetCommand;
exports.DeleteTrafficPolicyCommand = DeleteTrafficPolicyCommand;
exports.DeleteTrafficPolicyInstanceCommand = DeleteTrafficPolicyInstanceCommand;
exports.DeleteVPCAssociationAuthorizationCommand = DeleteVPCAssociationAuthorizationCommand;
exports.DisableHostedZoneDNSSECCommand = DisableHostedZoneDNSSECCommand;
exports.DisassociateVPCFromHostedZoneCommand = DisassociateVPCFromHostedZoneCommand;
exports.EnableHostedZoneDNSSECCommand = EnableHostedZoneDNSSECCommand;
exports.GetAccountLimitCommand = GetAccountLimitCommand;
exports.GetChangeCommand = GetChangeCommand;
exports.GetCheckerIpRangesCommand = GetCheckerIpRangesCommand;
exports.GetDNSSECCommand = GetDNSSECCommand;
exports.GetGeoLocationCommand = GetGeoLocationCommand;
exports.GetHealthCheckCommand = GetHealthCheckCommand;
exports.GetHealthCheckCountCommand = GetHealthCheckCountCommand;
exports.GetHealthCheckLastFailureReasonCommand = GetHealthCheckLastFailureReasonCommand;
exports.GetHealthCheckStatusCommand = GetHealthCheckStatusCommand;
exports.GetHostedZoneCommand = GetHostedZoneCommand;
exports.GetHostedZoneCountCommand = GetHostedZoneCountCommand;
exports.GetHostedZoneLimitCommand = GetHostedZoneLimitCommand;
exports.GetQueryLoggingConfigCommand = GetQueryLoggingConfigCommand;
exports.GetReusableDelegationSetCommand = GetReusableDelegationSetCommand;
exports.GetReusableDelegationSetLimitCommand = GetReusableDelegationSetLimitCommand;
exports.GetTrafficPolicyCommand = GetTrafficPolicyCommand;
exports.GetTrafficPolicyInstanceCommand = GetTrafficPolicyInstanceCommand;
exports.GetTrafficPolicyInstanceCountCommand = GetTrafficPolicyInstanceCountCommand;
exports.HealthCheckAlreadyExists = HealthCheckAlreadyExists$1;
exports.HealthCheckInUse = HealthCheckInUse$1;
exports.HealthCheckRegion = HealthCheckRegion;
exports.HealthCheckType = HealthCheckType;
exports.HealthCheckVersionMismatch = HealthCheckVersionMismatch$1;
exports.HostedZoneAlreadyExists = HostedZoneAlreadyExists$1;
exports.HostedZoneLimitType = HostedZoneLimitType;
exports.HostedZoneNotEmpty = HostedZoneNotEmpty$1;
exports.HostedZoneNotFound = HostedZoneNotFound$1;
exports.HostedZoneNotPrivate = HostedZoneNotPrivate$1;
exports.HostedZonePartiallyDelegated = HostedZonePartiallyDelegated$1;
exports.HostedZoneType = HostedZoneType;
exports.IncompatibleVersion = IncompatibleVersion$1;
exports.InsufficientCloudWatchLogsResourcePolicy = InsufficientCloudWatchLogsResourcePolicy$1;
exports.InsufficientDataHealthStatus = InsufficientDataHealthStatus;
exports.InvalidArgument = InvalidArgument$1;
exports.InvalidChangeBatch = InvalidChangeBatch$1;
exports.InvalidDomainName = InvalidDomainName$1;
exports.InvalidInput = InvalidInput$1;
exports.InvalidKMSArn = InvalidKMSArn$1;
exports.InvalidKeySigningKeyName = InvalidKeySigningKeyName$1;
exports.InvalidKeySigningKeyStatus = InvalidKeySigningKeyStatus$1;
exports.InvalidPaginationToken = InvalidPaginationToken$1;
exports.InvalidSigningStatus = InvalidSigningStatus$1;
exports.InvalidTrafficPolicyDocument = InvalidTrafficPolicyDocument$1;
exports.InvalidVPCId = InvalidVPCId$1;
exports.KeySigningKeyAlreadyExists = KeySigningKeyAlreadyExists$1;
exports.KeySigningKeyInParentDSRecord = KeySigningKeyInParentDSRecord$1;
exports.KeySigningKeyInUse = KeySigningKeyInUse$1;
exports.KeySigningKeyWithActiveStatusNotFound = KeySigningKeyWithActiveStatusNotFound$1;
exports.LastVPCAssociation = LastVPCAssociation$1;
exports.LimitsExceeded = LimitsExceeded$1;
exports.ListCidrBlocksCommand = ListCidrBlocksCommand;
exports.ListCidrCollectionsCommand = ListCidrCollectionsCommand;
exports.ListCidrLocationsCommand = ListCidrLocationsCommand;
exports.ListGeoLocationsCommand = ListGeoLocationsCommand;
exports.ListHealthChecksCommand = ListHealthChecksCommand;
exports.ListHostedZonesByNameCommand = ListHostedZonesByNameCommand;
exports.ListHostedZonesByVPCCommand = ListHostedZonesByVPCCommand;
exports.ListHostedZonesCommand = ListHostedZonesCommand;
exports.ListQueryLoggingConfigsCommand = ListQueryLoggingConfigsCommand;
exports.ListResourceRecordSetsCommand = ListResourceRecordSetsCommand;
exports.ListReusableDelegationSetsCommand = ListReusableDelegationSetsCommand;
exports.ListTagsForResourceCommand = ListTagsForResourceCommand;
exports.ListTagsForResourcesCommand = ListTagsForResourcesCommand;
exports.ListTrafficPoliciesCommand = ListTrafficPoliciesCommand;
exports.ListTrafficPolicyInstancesByHostedZoneCommand = ListTrafficPolicyInstancesByHostedZoneCommand;
exports.ListTrafficPolicyInstancesByPolicyCommand = ListTrafficPolicyInstancesByPolicyCommand;
exports.ListTrafficPolicyInstancesCommand = ListTrafficPolicyInstancesCommand;
exports.ListTrafficPolicyVersionsCommand = ListTrafficPolicyVersionsCommand;
exports.ListVPCAssociationAuthorizationsCommand = ListVPCAssociationAuthorizationsCommand;
exports.NoSuchChange = NoSuchChange$1;
exports.NoSuchCidrCollectionException = NoSuchCidrCollectionException$1;
exports.NoSuchCidrLocationException = NoSuchCidrLocationException$1;
exports.NoSuchCloudWatchLogsLogGroup = NoSuchCloudWatchLogsLogGroup$1;
exports.NoSuchDelegationSet = NoSuchDelegationSet$1;
exports.NoSuchGeoLocation = NoSuchGeoLocation$1;
exports.NoSuchHealthCheck = NoSuchHealthCheck$1;
exports.NoSuchHostedZone = NoSuchHostedZone$1;
exports.NoSuchKeySigningKey = NoSuchKeySigningKey$1;
exports.NoSuchQueryLoggingConfig = NoSuchQueryLoggingConfig$1;
exports.NoSuchTrafficPolicy = NoSuchTrafficPolicy$1;
exports.NoSuchTrafficPolicyInstance = NoSuchTrafficPolicyInstance$1;
exports.NotAuthorizedException = NotAuthorizedException$1;
exports.PriorRequestNotComplete = PriorRequestNotComplete$1;
exports.PublicZoneVPCAssociation = PublicZoneVPCAssociation$1;
exports.QueryLoggingConfigAlreadyExists = QueryLoggingConfigAlreadyExists$1;
exports.RRType = RRType;
exports.ResettableElementName = ResettableElementName;
exports.ResourceRecordSetFailover = ResourceRecordSetFailover;
exports.ResourceRecordSetRegion = ResourceRecordSetRegion;
exports.ReusableDelegationSetLimitType = ReusableDelegationSetLimitType;
exports.Route53 = Route53;
exports.Route53Client = Route53Client;
exports.Route53ServiceException = Route53ServiceException$1;
exports.Statistic = Statistic;
exports.TagResourceType = TagResourceType;
exports.TestDNSAnswerCommand = TestDNSAnswerCommand;
exports.ThrottlingException = ThrottlingException$1;
exports.TooManyHealthChecks = TooManyHealthChecks$1;
exports.TooManyHostedZones = TooManyHostedZones$1;
exports.TooManyKeySigningKeys = TooManyKeySigningKeys$1;
exports.TooManyTrafficPolicies = TooManyTrafficPolicies$1;
exports.TooManyTrafficPolicyInstances = TooManyTrafficPolicyInstances$1;
exports.TooManyTrafficPolicyVersionsForCurrentPolicy = TooManyTrafficPolicyVersionsForCurrentPolicy$1;
exports.TooManyVPCAssociationAuthorizations = TooManyVPCAssociationAuthorizations$1;
exports.TrafficPolicyAlreadyExists = TrafficPolicyAlreadyExists$1;
exports.TrafficPolicyInUse = TrafficPolicyInUse$1;
exports.TrafficPolicyInstanceAlreadyExists = TrafficPolicyInstanceAlreadyExists$1;
exports.UpdateHealthCheckCommand = UpdateHealthCheckCommand;
exports.UpdateHostedZoneCommentCommand = UpdateHostedZoneCommentCommand;
exports.UpdateTrafficPolicyCommentCommand = UpdateTrafficPolicyCommentCommand;
exports.UpdateTrafficPolicyInstanceCommand = UpdateTrafficPolicyInstanceCommand;
exports.VPCAssociationAuthorizationNotFound = VPCAssociationAuthorizationNotFound$1;
exports.VPCAssociationNotFound = VPCAssociationNotFound$1;
exports.VPCRegion = VPCRegion;
exports.paginateListCidrBlocks = paginateListCidrBlocks;
exports.paginateListCidrCollections = paginateListCidrCollections;
exports.paginateListCidrLocations = paginateListCidrLocations;
exports.paginateListHealthChecks = paginateListHealthChecks;
exports.paginateListHostedZones = paginateListHostedZones;
exports.paginateListQueryLoggingConfigs = paginateListQueryLoggingConfigs;
exports.waitForResourceRecordSetsChanged = waitForResourceRecordSetsChanged;
exports.waitUntilResourceRecordSetsChanged = waitUntilResourceRecordSetsChanged;
